---
title: "P76156 Patient Report"
author: "SYGNOMICS"
date: "12/04/2021"
output:
  html_document:
    df_print: paged
  always_allow_html: yes
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
  options(width = 1200)

```

```{r libraries, echo=FALSE, message=FALSE,warning=FALSE}
library(tidyverse)
library(DT)
library(icons)
```

```{r load data, echo=FALSE, message=FALSE,warning=FALSE}
# Load patient data
patient_id = 'P76156_3'
filename <- paste("output/",patient_id, "/", patient_id, "_regulon_drug_therapy_activity.csv",sep="")
        network_info <-
            read.csv(
                filename,
            )

```

```{r patient meta, results='asis', echo=FALSE, message=FALSE,warning=FALSE}
        tmp6 <- tibble(`Patient ID` = patient_id,
                  `Age` = "",
                  `Sex` = "",
                  `MGMT Methylation` = "Methylated",
                  `ExomeSeq` = paste("Yes"),
                  `RNASeq` = paste("Yes"),
                  `CNV` = paste("Yes"),
                  `Profiling Type` = "Unknown"
                  )
     datatable(tmp6, options = list(dom=''))


```

### Drug Report Summary based on Patient's active risk-associated transcriptional programs.
```{r drug summary, results='asis', echo=F, warning=F, message=F}

for(drug in unique(network_info$Drug)){
  cat(sep="","1. Drug: **", drug, "**  \n")
  
    activity.counts <- network_info %>%
    filter(Drug == drug) %>%
    separate(Regulon.Activity.Summary, sep=" ", into=c("i1","Over","i3","Under","i5","All"), remove = F) %>%
    select("Over","Under","All")
  
  target.list <- network_info %>%
    filter(Drug == drug) %>%
    pull(Target.Gene.s.) %>%
    unique()
  
  rmsts <- network_info %>%
    filter(Drug == drug) %>%
    pull(Mean.RMST.Regulon)
  
  gbmtrials <- network_info %>%
    filter(Drug == drug) %>%
    pull(DrugTrialPhaseGBM) %>%
    unique()
  
   anticancerPhaseIVs <- network_info %>%
    filter(Drug == drug) %>%
    select(Other.FDA.Appr.) %>%
    unique() %>%
    mutate(Other.FDA.Appr. = if_else(Other.FDA.Appr. == TRUE, paste("**Phase IV** trial for some other cancer(s)"), paste(" **NOT** in Phase IV trials for any cancer."))) %>%
     pull()
   
   evidence.list <- drugs %>%
    filter(Drug == drug) %>%
    select(Evidence) %>%
    unique() %>%
    mutate(`Evidence` = if_else(Evidence == 0, "", Evidence)) %>%
    pull()
   evidence.sep <- paste(str_split(evidence.list, pattern = ",")[[1]], collapse=", ")

   cat(sep = "","Evidence: **", length(target.list),  "** risk-associated regulons containing the target(s), ", paste("**",paste(target.list, collapse=","),"**", sep=""), " of the drug **", drug, "** are overactive in this patient with max RMST value of **", max(rmsts), "**.\n")
   cat(sep="", drug, " has **", gbmtrials, "** clinical trials for GBM and ", "is in ", anticancerPhaseIVs, "  \n" )
   if(evidence.sep != ""){
     cat(sep="", "+ Additional Evidence(s): ", evidence.sep, "  \n")
   }
   
  cat("\n\n")
  cat("\n")
}

```


### Drug Report Summary based on Patient's Mutations
```{r mutation drug summary, results='asis', echo=F, warning=F, message=F}

drugs <- network_info %>%
  filter(TL1961DB85RegulonActivity == 1 & Miner_Target_Type == 'MutationGene') %>%
  select(Drug, MutationSymbol, TargetSymbol, RegulatorID, Regulon_ID,RegulatorSymbol, RMST_diff_UnderActiveMinusOverActiveMGMTMeth, DrugTrialPhaseGBM,IfAntiCancerPhaseIV, MutationRegulatorEdge) %>%
  unique()
  
for(mutation in unique(drugs$MutationSymbol)){
  cat(sep="","1. Mutation: **", mutation, "**  \n")
  
  target.drugs <- drugs %>%
    filter(MutationSymbol == mutation) %>%
    pull(Drug) %>%
    unique()
   
  target.regulators <- drugs %>%
    filter(MutationSymbol == mutation) %>%
    pull(RegulatorSymbol) %>%
    unique()
   
  
  target.regulons <- drugs %>%
    filter(MutationSymbol == mutation) %>%
    pull(Regulon_ID) %>%
    unique()
  
  target.list <- drugs %>%
    filter(MutationSymbol == mutation) %>%
    pull(TargetSymbol) %>%
    unique()
  
  rmsts <- drugs %>%
    filter(MutationSymbol == mutation) %>%
    pull(RMST_diff_UnderActiveMinusOverActiveMGMTMeth)
  
  gbmtrials <- drugs %>%
    filter(MutationSymbol == mutation) %>%
    pull(DrugTrialPhaseGBM) %>%
    unique()
  
  mutation2regulators <- drugs %>%
    filter(MutationSymbol == mutation) %>%
    select(MutationRegulatorEdge) %>%
    unique() %>%
    mutate(`MutationRegulatorEdge` = if_else(MutationRegulatorEdge == "1", "activates", "suppresses")) %>%
    pull()
  
   anticancerPhaseIVs <- drugs %>%
    filter(MutationSymbol == mutation) %>%
    select(IfAntiCancerPhaseIV) %>%
    unique() %>%
    mutate(`IfAntiCancerPhaseIV` = if_else(IfAntiCancerPhaseIV == "AntiCancerPhaseIV", paste("**Phase IV** trial for some other cancer(s)"), paste(" **NOT** in Phase IV trials for any cancer."))) %>%
     pull()

   cat(sep = "","Evidence: Patient has **",mutation, "** mutation that **" ,mutation2regulators,"** regulator, ",target.regulators, " of **", length(target.regulons),  "** active risk-associated regulons containing the target(s), ", paste("**",paste(target.list, collapse=","),"**", sep=""), " of the drug **", paste(drug, collapse=","), "** with max RMST value of **", max(rmsts), "**.\n")
  # cat(sep="", drug, " has **", paste(gbmtrials, collapse=","), "** clinical trials for GBM.  \n" )
  cat("  \n\n")
  cat("  \n")
}

```


### Summary Table
```{r active regulons filter, echo=FALSE, message=FALSE,warning=FALSE}
network_table1 <- network_info %>%
  filter(TL1961DB85RegulonActivity == 1 & Miner_Target_Type == 'RegulonGene') %>%
  group_by(TargetSymbol) %>%
  rename(Drug.Target = TargetSymbol) %>%
  summarize(Drugs = paste(unique(Drug), collapse =", "),
            Active.Regulons = paste(unique(Regulon_ID), collapse=", "),
            Active.Regulon.TFs = paste(unique(RegulatorSymbol), collapse =", "),
            Max.RMST = max(unique(RMST_diff_UnderActiveMinusOverActiveMGMTMeth)),
            Mean.RMST = mean(unique(RMST_diff_UnderActiveMinusOverActiveMGMTMeth)),
            GBM.Trials = DrugTrialPhaseGBM,
            ) %>%
  unique()

#print(htmltools::tagList(datatable(network_table1,rownames = F,escape = F, caption = paste("Summary of Rank Ordered Drugs ", " \n",sep=""))))
  network_table1 %>% datatable(
                escape = F,
                filter = "top",
                rownames = F,
                selection = "single",
                style = "bootstrap4",
                extensions = c("Buttons"),
                callback = JS('table.page(3).draw(false);'),
                options = list(
                  headerCallback = DT::JS(
    "function(thead) {",
    "  $(thead).css('font-size', '80%');",
    "}"
  ),
                            defaultStyle.fontSize='10%',
                    buttons =
                        list(
                            'pageLength',
                            'colvis',
                            list(extend='excel', title="gbmSYGNAL Details"),
                            list(
                                extend = 'pdf',
                                title="gbmSYGNAL Network Details",
                                pageSize = 'letter',
                                orientation = 'landscape',
                                filename = 'SYGNAL_Network_Details',
                                customize = JS("function(doc){
                                  doc.styles.tableHeader.color='white';
                                  doc.defaultStyle.alignment = 'left';
                                  doc.styles.tableHeader.alignment = 'left';
                                  doc.pageMargins = [10,10,10,10];
                                  doc.defaultStyle.fontSize = 7;
                                  doc.styles.tableHeader.fontSize = 7;
                                  doc.styles.title.fontSize = 9;
                                  
                                  }"
                                )
                            )
                        ),
                    scrollX = TRUE,
                    # allow user to scroll wide tables horizontally
                    stateSave = FALSE,
                    
                    # default column search strings and global search string
                    target = "row",
                    dom =
                        'Britp'
                    
                )
            ) %>%
    DT::formatStyle(columns = colnames(network_table1),fontSize='80%')



```
