---
title: '`r paste0(patient_id, " SYGNOMICS Report")`'
#author: "SYGNOMICS"
date: "01/09/2022"
output:
  html_document:
    df_print: paged
  always_allow_html: yes
  pdf_document: TRUE
  width: "100%"
  css: style.css
  theme: flatly
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
  options(width = 1200)
  options(knitr.table.format = "html") 

```

```{r libraries, echo=FALSE, message=FALSE,warning=FALSE}
library(tidyverse)
library(DT)
library(grid)
library(gridExtra)
library(kableExtra)
library(here)
library(jsonlite)
library(plyr)
library(purrr)
library(clusterProfiler)
library(msigdbr)
library(enrichplot)
library(ReactomePA)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(data.table)
library(tictoc)
library(fs)
source("../code/utilities_drug.R")
source("../code/patient_get_network_details.R")
source("../code/01_analyze_patient.R")
source("../code/format_target_data.R")
source("../code/patient_plots.R")
#library(icons)
```


```{r soc_drugs, echo=FALSE, message=FALSE,warning=FALSE}
soc_drugs <- c("TEMOZOLOMIDE","CARMUSTINE","BEVACIZUMAB","LOMUSTINE", "NIMUSTINE","FOTEMUSTINE")
```


```{r patient meta, results='asis', echo=FALSE, message=FALSE,warning=FALSE}
        tmp6 <- tibble(`Patient ID` = patient_id,
                  `Age` = "",
                  `Sex` = "",
                  `MGMT Methylation` = "Methylated",
                  `ExomeSeq` = paste("Yes"),
                  `RNASeq` = paste("Yes"),
                  `CNV` = paste("Yes"),
                  `Profiling Type` = "Unknown"
                  )
     datatable(tmp6, options = list(dom=''))
```


```{r include = FALSE}
# Why, oh why do I need this chunk?
#datatable(NULL)
DT::datatable(matrix())
```


# SOC Drugs
```{r soc_drug_summary, results='asis', echo=F, warning=F, message=F}
cat(sep="", '<a name="analysis_soc"></a>')
count = 0
network_info_soc <- network_info %>%
  filter(Drug %in% soc_drugs)

for(target in unique(network_info_soc$`Target Gene(s)`)){
  count = count + 1
 format_target_data(target, count=count, network_info=network_info_soc, clinical_trials, evidence)
}

```

# Drugs in GBM Trials
```{r drugs_gbmtrial, results='asis', echo=F, warning=F, message=F}
cat(sep="", '<a name="analysis_gbmtrials"></a>')
count = 0
network_info_gbmtrials <- network_info %>%
  filter(!is.na(max_glioblastoma.multiforme_phase)) %>%
  filter(!(Drug %in% soc_drugs))
  
for(target in unique(network_info_gbmtrials$`Target Gene(s)`)){
  count = count + 1
 format_target_data(target, count=count, network_info=network_info_gbmtrials, clinical_trials, evidence)
}

```

# FDA Approved Drugs
```{r fda_approved, results='asis', echo=F, warning=F, message=F}
cat(sep="", '<a name="analysis_fda"></a>')
count = 0
network_info_fda <- network_info %>%
  filter(`Other FDA Appr.` == TRUE) %>%
  filter(is.na(max_glioblastoma.multiforme_phase)) %>%
  filter(!(Drug %in% soc_drugs))
 
  
for(target in unique(network_info_fda$`Target Gene(s)`)){
  count = count + 1
 format_target_data(target, count=count, network_info=network_info_fda, clinical_trials, evidence)
}
warnings()

```
