<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="SYGNOMICS" />

<meta name="date" content="2021-12-04" />

<title>P76156 Patient Report</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.1/jquery-3.5.1.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.19/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/dt-core-bootstrap4-1.10.20/css/dataTables.bootstrap4.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-bootstrap4-1.10.20/css/dataTables.bootstrap4.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-bootstrap4-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="site_libs/dt-core-bootstrap4-1.10.20/js/dataTables.bootstrap4.min.js"></script>
<script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-bootstrap4-1.10.20/css/buttons.bootstrap4.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-bootstrap4-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-bootstrap4-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-bootstrap4-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-bootstrap4-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-bootstrap4-1.10.20/js/buttons.print.min.js"></script>
<script src="site_libs/dt-ext-buttons-bootstrap4-1.10.20/js/buttons.bootstrap4.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">XCures</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">P76156 Patient Report</h1>
<h4 class="author">SYGNOMICS</h4>
<h4 class="date">12/04/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>XCures/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211122code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211122)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211122code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211122)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsturkarslanGbmMINERtree064735f65ad77d8640bda848e274c85103f4da52targetblank064735fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sturkarslan/GbmMINER/tree/064735f65ad77d8640bda848e274c85103f4da52" target="_blank">064735f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsturkarslanGbmMINERtree064735f65ad77d8640bda848e274c85103f4da52targetblank064735fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sturkarslan/GbmMINER/tree/064735f65ad77d8640bda848e274c85103f4da52" target="_blank">064735f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    PD-GSC/.DS_Store
    Ignored:    PD-GSC/.Rhistory
    Ignored:    PD-GSC/.Rproj.user/
    Ignored:    PD-GSC/output/.DS_Store
    Ignored:    XCures/.Rhistory
    Ignored:    XCures/.Rproj.user/
    Ignored:    XCures/output/TL-18-31DCF0_all_regulon_activity.csv/
    Ignored:    output/01_SN505-Neurosphere-P2/

Untracked files:
    Untracked:  PD-GSC/.Rprofile
    Untracked:  PD-GSC/.gitattributes
    Untracked:  PD-GSC/.gitignore
    Untracked:  PD-GSC/AUC_Program_Activity.pdf
    Untracked:  PD-GSC/AUC_RegulonActivity.pdf
    Untracked:  PD-GSC/AUC_Regulon_Activity.pdf
    Untracked:  PD-GSC/README.md
    Untracked:  PD-GSC/_workflowr.yml
    Untracked:  PD-GSC/analysis/
    Untracked:  PD-GSC/code/
    Untracked:  PD-GSC/data/
    Untracked:  PD-GSC/docs/
    Untracked:  PD-GSC/output/17_SN649-Adherent-P5/
    Untracked:  PD-GSC/output/18_SN655-Neurosphere-P3/
    Untracked:  PD-GSC/output/19_SN667-Neurosphere-P2/
    Untracked:  PD-GSC/output/20_SN674-Neurosphere-P2/
    Untracked:  PD-GSC/output/21_SN677-Neurosphere-P2/
    Untracked:  PD-GSC/output/22_SN678-Adherent-P3/
    Untracked:  PD-GSC/output/23_SN683-Neurosphere-P3/
    Untracked:  PD-GSC/output/26_SN720-Adherent-p2/
    Untracked:  PD-GSC/output/27_SN735-Neurosphere-P9/
    Untracked:  PD-GSC/output/28_SN747-Neurosphere-P4/
    Untracked:  PD-GSC/output/29_SN753-Neurosphere-P3/
    Untracked:  PD-GSC/output/30_SN756-Neurosphere-P6/
    Untracked:  PD-GSC/output/31_SN758-Neurosphere-P6/
    Untracked:  PD-GSC/output/32_SN767-Neurosphere-P5/
    Untracked:  PD-GSC/output/33_SN787-Neurosphere-P6/
    Untracked:  PD-GSC/output/34_SN788-Neurosphere-P5/
    Untracked:  PD-GSC/output/35_SN799-Neurosphere-P3/
    Untracked:  PD-GSC/output/36_SN804-Neurosphere-P2/
    Untracked:  PD-GSC/output/37_SN810-Neurosphere-P5/
    Untracked:  PD-GSC/output/38_SN812-Neurosphere-P6/
    Untracked:  PD-GSC/output/39_SN820-Neurosphere-P5/
    Untracked:  PD-GSC/output/README.md
    Untracked:  XCures/code/calculate_regulon_activity_cohort.R
    Untracked:  _workflowr.yml
    Untracked:  data/MINER_MicroLowessRNATMM.08.24.2020-ST/
    Untracked:  data/MINER_MicroLowessRNATMM.08.24.2020/.DS_Store
    Untracked:  data/README.md

Unstaged changes:
    Modified:   .DS_Store
    Deleted:    PD-GSC/.ipynb_checkpoints/regulonActivity-PDGSC-checkpoint.ipynb
    Deleted:    PD-GSC/PreliminaryDrugTableForTherapyPrioritization.txt
    Deleted:    PD-GSC/Program_Activity_01_SN505-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_02_SN508-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_03_SN515-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_04_SN517-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_05_SN521-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_06_SN528-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_07_SN529-Neurosphere-P1.csv
    Deleted:    PD-GSC/Program_Activity_08_SN531-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_10_SN534-Neurosphere-P1.csv
    Deleted:    PD-GSC/Program_Activity_12_SN553-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_13_SN575-Neurosphere-P4.csv
    Deleted:    PD-GSC/Program_Activity_14_SN579-Neurosphere-P1.csv
    Deleted:    PD-GSC/Program_Activity_15_SN598-Neurosphere-P5.csv
    Deleted:    PD-GSC/Program_Activity_16_SN601-Neurosphere-p5.csv
    Deleted:    PD-GSC/Program_Activity_17_SN649-Adherent-P5.csv
    Deleted:    PD-GSC/Program_Activity_18_SN655-Neurosphere-P3.csv
    Deleted:    PD-GSC/Program_Activity_19_SN667-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_20_SN674-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_21_SN677-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_22_SN678-Adherent-P3.csv
    Deleted:    PD-GSC/Program_Activity_23_SN683-Neurosphere-P3.csv
    Deleted:    PD-GSC/Program_Activity_26_SN720-Adherent-p2.csv
    Deleted:    PD-GSC/Program_Activity_27_SN735-Neurosphere-P9.csv
    Deleted:    PD-GSC/Program_Activity_28_SN747-Neurosphere-P4.csv
    Deleted:    PD-GSC/Program_Activity_29_SN753-Neurosphere-P3.csv
    Deleted:    PD-GSC/Program_Activity_30_SN756-Neurosphere-P6.csv
    Deleted:    PD-GSC/Program_Activity_31_SN758-Neurosphere-P6.csv
    Deleted:    PD-GSC/Program_Activity_32_SN767-Neurosphere-P5.csv
    Deleted:    PD-GSC/Program_Activity_33_SN787-Neurosphere-P6.csv
    Deleted:    PD-GSC/Program_Activity_34_SN788-Neurosphere-P5.csv
    Deleted:    PD-GSC/Program_Activity_35_SN799-Neurosphere-P3.csv
    Deleted:    PD-GSC/Program_Activity_36_SN804-Neurosphere-P2.csv
    Deleted:    PD-GSC/Program_Activity_37_SN810-Neurosphere-P5.csv
    Deleted:    PD-GSC/Program_Activity_38_SN812-Neurosphere-P6.csv
    Deleted:    PD-GSC/Program_Activity_39_SN820-Neurosphere-P5.csv
    Deleted:    PD-GSC/Regulon_Activity_01_SN505-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_02_SN508-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_03_SN515-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_04_SN517-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_05_SN521-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_06_SN528-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_07_SN529-Neurosphere-P1.csv
    Deleted:    PD-GSC/Regulon_Activity_08_SN531-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_10_SN534-Neurosphere-P1.csv
    Deleted:    PD-GSC/Regulon_Activity_12_SN553-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_13_SN575-Neurosphere-P4.csv
    Deleted:    PD-GSC/Regulon_Activity_14_SN579-Neurosphere-P1.csv
    Deleted:    PD-GSC/Regulon_Activity_15_SN598-Neurosphere-P5.csv
    Deleted:    PD-GSC/Regulon_Activity_16_SN601-Neurosphere-p5.csv
    Deleted:    PD-GSC/Regulon_Activity_17_SN649-Adherent-P5.csv
    Deleted:    PD-GSC/Regulon_Activity_18_SN655-Neurosphere-P3.csv
    Deleted:    PD-GSC/Regulon_Activity_19_SN667-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_20_SN674-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_21_SN677-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_22_SN678-Adherent-P3.csv
    Deleted:    PD-GSC/Regulon_Activity_23_SN683-Neurosphere-P3.csv
    Deleted:    PD-GSC/Regulon_Activity_26_SN720-Adherent-p2.csv
    Deleted:    PD-GSC/Regulon_Activity_27_SN735-Neurosphere-P9.csv
    Deleted:    PD-GSC/Regulon_Activity_28_SN747-Neurosphere-P4.csv
    Deleted:    PD-GSC/Regulon_Activity_29_SN753-Neurosphere-P3.csv
    Deleted:    PD-GSC/Regulon_Activity_30_SN756-Neurosphere-P6.csv
    Deleted:    PD-GSC/Regulon_Activity_31_SN758-Neurosphere-P6.csv
    Deleted:    PD-GSC/Regulon_Activity_32_SN767-Neurosphere-P5.csv
    Deleted:    PD-GSC/Regulon_Activity_33_SN787-Neurosphere-P6.csv
    Deleted:    PD-GSC/Regulon_Activity_34_SN788-Neurosphere-P5.csv
    Deleted:    PD-GSC/Regulon_Activity_35_SN799-Neurosphere-P3.csv
    Deleted:    PD-GSC/Regulon_Activity_36_SN804-Neurosphere-P2.csv
    Deleted:    PD-GSC/Regulon_Activity_37_SN810-Neurosphere-P5.csv
    Deleted:    PD-GSC/Regulon_Activity_38_SN812-Neurosphere-P6.csv
    Deleted:    PD-GSC/Regulon_Activity_39_SN820-Neurosphere-P5.csv
    Deleted:    PD-GSC/__pycache__/miner_py3_kk.cpython-37.pyc
    Deleted:    PD-GSC/miner_py3_kk.py
    Modified:   PD-GSC/output/01_SN505-Neurosphere-P2/01_SN505-Neurosphere-P2_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/02_SN508-Neurosphere-P2/02_SN508-Neurosphere-P2_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/03_SN515-Neurosphere-P2/03_SN515-Neurosphere-P2_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/04_SN517-Neurosphere-P2/04_SN517-Neurosphere-P2_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/05_SN521-Neurosphere-P2/05_SN521-Neurosphere-P2_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/06_SN528-Neurosphere-P2/06_SN528-Neurosphere-P2_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/07_SN529-Neurosphere-P1/07_SN529-Neurosphere-P1_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/08_SN531-Neurosphere-P2/08_SN531-Neurosphere-P2_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/10_SN534-Neurosphere-P1/10_SN534-Neurosphere-P1_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/12_SN553-Neurosphere-P2/12_SN553-Neurosphere-P2_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/13_SN575-Neurosphere-P4/13_SN575-Neurosphere-P4_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/14_SN579-Neurosphere-P1/14_SN579-Neurosphere-P1_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/15_SN598-Neurosphere-P5/15_SN598-Neurosphere-P5_drug_therapy_activity_IC50.csv
    Modified:   PD-GSC/output/16_SN601-Neurosphere-p5/16_SN601-Neurosphere-p5_drug_therapy_activity_IC50.csv
    Deleted:    PD-GSC/regulonActivity-PDGSC.ipynb
    Deleted:    PD-GSC/regulonActivitySummary.txt
    Deleted:    PD-GSC/unique_GBM_drugs.csv
    Modified:   XCures/analysis/P76156_Analysis.Rmd
    Modified:   XCures/code/patient_plots.R
    Modified:   data/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>XCures/analysis/P76156_Analysis.Rmd</code>) and HTML (<code>XCures/docs/P76156_Analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sturkarslan/GbmMINER/blob/064735f65ad77d8640bda848e274c85103f4da52/XCures/analysis/P76156_Analysis.Rmd" target="_blank">064735f</a>
</td>
<td>
sturkarslan
</td>
<td>
2021-12-06
</td>
<td>
Added potting and notebook for patient report
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="htmlwidget-7c9bc05e69bc1b95f864" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7c9bc05e69bc1b95f864">{"x":{"filter":"none","vertical":false,"data":[["1"],["P76156_3"],[""],[""],["Methylated"],["Yes"],["Yes"],["Yes"],["Unknown"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Patient ID<\/th>\n      <th>Age<\/th>\n      <th>Sex<\/th>\n      <th>MGMT Methylation<\/th>\n      <th>ExomeSeq<\/th>\n      <th>RNASeq<\/th>\n      <th>CNV<\/th>\n      <th>Profiling Type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"","order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
<div id="drug-report-summary-based-on-patients-active-risk-associated-transcriptional-programs." class="section level3">
<h3>Drug Report Summary based on Patient’s active risk-associated transcriptional programs.</h3>
<ol style="list-style-type: decimal">
<li>Target: <strong>VEGFA</strong><br />
Drug(s): <strong>BEVACIZUMAB,RANIBIZUMAB</strong><br />
Evidence: <strong>4</strong> risk-associated regulons containing the target(s), <strong>VEGFA</strong> of the drug(s) <strong>BEVACIZUMAB,RANIBIZUMAB</strong> are overactive in this patient with max RMST value of <strong>280.65</strong>. BEVACIZUMAB,RANIBIZUMAB has (<strong>Phase 2,None</strong>) clinical trials for GBM and is in <strong>Phase IV</strong> trial for some other cancer(s)<br />
</li>
</ol>
<ul>
<li>Hallmark(s): HALLMARK_HYPOXIA, HALLMARK_TNFA_SIGNALING_VIA_NFKB<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-1.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<ol style="list-style-type: decimal">
<li><p>Target: <strong>RAF1</strong><br />
Drug(s): <strong>SORAFENIB</strong><br />
Evidence: <strong>1</strong> risk-associated regulons containing the target(s), <strong>RAF1</strong> of the drug(s) <strong>SORAFENIB</strong> are overactive in this patient with max RMST value of <strong>-247.54</strong>. SORAFENIB has (<strong>Phase 2</strong>) clinical trials for GBM and is in <strong>Phase IV</strong> trial for some other cancer(s)<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-2.png" width="672" style="display: block; margin: auto;" /></p></li>
<li><p>Target: <strong>HRH1</strong><br />
Drug(s): <strong>DEXCHLORPHENIRAMINE,HISTAMINE,CYPROHEPTADINE</strong><br />
Evidence: <strong>15</strong> risk-associated regulons containing the target(s), <strong>HRH1</strong> of the drug(s) <strong>DEXCHLORPHENIRAMINE,HISTAMINE,CYPROHEPTADINE</strong> are overactive in this patient with max RMST value of <strong>290.92</strong>. DEXCHLORPHENIRAMINE,HISTAMINE,CYPROHEPTADINE has (<strong>None,None,Early Phase 1</strong>) clinical trials for GBM and is in <strong>Phase IV</strong> trial for some other cancer(s) <strong>NOT</strong> in Phase IV trials for any cancer.<br />
</p></li>
</ol>
<ul>
<li>Hallmark(s): HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION, HALLMARK_KRAS_SIGNALING_UP, HALLMARK_HYPOXIA<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-3.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<ol style="list-style-type: decimal">
<li><p>Target: <strong>CHRNB1</strong><br />
Drug(s): <strong>ROCURONIUM</strong><br />
Evidence: <strong>3</strong> risk-associated regulons containing the target(s), <strong>CHRNB1</strong> of the drug(s) <strong>ROCURONIUM</strong> are overactive in this patient with max RMST value of <strong>214.29</strong>. ROCURONIUM has (<strong>None</strong>) clinical trials for GBM and is in <strong>Phase IV</strong> trial for some other cancer(s)<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-4.png" width="672" style="display: block; margin: auto;" /></p></li>
<li><p>Target: <strong>OXTR</strong><br />
Drug(s): <strong>CARBETOCIN,OXYTOCIN</strong><br />
Evidence: <strong>3</strong> risk-associated regulons containing the target(s), <strong>OXTR</strong> of the drug(s) <strong>CARBETOCIN,OXYTOCIN</strong> are overactive in this patient with max RMST value of <strong>236.43</strong>. CARBETOCIN,OXYTOCIN has (<strong>None,None</strong>) clinical trials for GBM and is in <strong>Phase IV</strong> trial for some other cancer(s)<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-5.png" width="672" style="display: block; margin: auto;" /></p></li>
<li><p>Target: <strong>NR3C1</strong><br />
Drug(s): <strong>DEXAMETHASONE,PREDNISONE,TRIAMCINOLONE,BUDESONIDE,HYDROCORTISONE,MIFEPRISTONE,PREDNISOLONE</strong><br />
Evidence: <strong>0</strong> risk-associated regulons containing the target(s), <strong>NR3C1</strong> of the drug(s) <strong>DEXAMETHASONE,PREDNISONE,TRIAMCINOLONE,BUDESONIDE,HYDROCORTISONE,MIFEPRISTONE,PREDNISOLONE</strong> are overactive in this patient with max RMST value of <strong>239.18</strong>. DEXAMETHASONE,PREDNISONE,TRIAMCINOLONE,BUDESONIDE,HYDROCORTISONE,MIFEPRISTONE,PREDNISOLONE has (<strong>Phase 2,Phase 2,Phase 2,None,None,None,None</strong>) clinical trials for GBM and is in <strong>Phase IV</strong> trial for some other cancer(s)<br />
</p></li>
</ol>
<ul>
<li>Hallmark(s): NA<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-6.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<ol style="list-style-type: decimal">
<li><p>Target: <strong>WEE1</strong><br />
Drug(s): <strong>ADAVOSERTIB</strong><br />
Evidence: <strong>2</strong> risk-associated regulons containing the target(s), <strong>WEE1</strong> of the drug(s) <strong>ADAVOSERTIB</strong> are overactive in this patient with max RMST value of <strong>245.07</strong>. ADAVOSERTIB has (<strong>Phase 1</strong>) clinical trials for GBM and is in <strong>NOT</strong> in Phase IV trials for any cancer.<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-7.png" width="672" style="display: block; margin: auto;" /></p></li>
<li><p>Target: <strong>HRH1,SCN3A,SCN5A,SCN9A</strong><br />
Drug(s): <strong>ORPHENADRINE</strong><br />
Evidence: <strong>15</strong> risk-associated regulons containing the target(s), <strong>HRH1,SCN3A,SCN5A,SCN9A</strong> of the drug(s) <strong>ORPHENADRINE</strong> are overactive in this patient with max RMST value of <strong>139.27</strong>. ORPHENADRINE has (<strong>None</strong>) clinical trials for GBM and is in <strong>Phase IV</strong> trial for some other cancer(s)<br />
</p></li>
</ol>
<ul>
<li>Hallmark(s): HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION, HALLMARK_KRAS_SIGNALING_UP, HALLMARK_HYPOXIA<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-8.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<ol style="list-style-type: decimal">
<li>Target: <strong>GPD2,NDUFA4L2,NDUFC1</strong><br />
Drug(s): <strong>METFORMIN</strong><br />
Evidence: <strong>1</strong> risk-associated regulons containing the target(s), <strong>GPD2,NDUFA4L2,NDUFC1</strong> of the drug(s) <strong>METFORMIN</strong> are overactive in this patient with max RMST value of <strong>44.36</strong>. METFORMIN has (<strong>Phase 2</strong>) clinical trials for GBM and is in <strong>Phase IV</strong> trial for some other cancer(s)<br />
</li>
</ol>
<ul>
<li>Hallmark(s): HALLMARK_MYC_TARGETS_V1, HALLMARK_UNFOLDED_PROTEIN_RESPONSE<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-9.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<ol style="list-style-type: decimal">
<li>Target: <strong>KCNJ8</strong><br />
Drug(s): <strong>DIAZOXIDE</strong><br />
Evidence: <strong>0</strong> risk-associated regulons containing the target(s), <strong>KCNJ8</strong> of the drug(s) <strong>DIAZOXIDE</strong> are overactive in this patient with max RMST value of <strong>281.78</strong>. DIAZOXIDE has (<strong>None</strong>) clinical trials for GBM and is in <strong>Phase IV</strong> trial for some other cancer(s)<br />
</li>
</ol>
<ul>
<li>Hallmark(s): NA<br />
<img src="figure/P76156_Analysis.Rmd/drug%20summary-10.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<!-- ### Drug Report Summary based on Patient's Mutations -->
<!-- ```{r mutation drug summary, results='asis', echo=F, warning=F, message=F} -->
<!-- drugs <- network_info %>% -->
<!--   filter(TL1961DB85RegulonActivity == 1 & Miner_Target_Type == 'MutationGene') %>% -->
<!--   select(Drug, MutationSymbol, TargetSymbol, RegulatorID, Regulon_ID,RegulatorSymbol, RMST_diff_UnderActiveMinusOverActiveMGMTMeth, DrugTrialPhaseGBM,IfAntiCancerPhaseIV, MutationRegulatorEdge) %>% -->
<!--   unique() -->
<!-- for(mutation in unique(drugs$MutationSymbol)){ -->
<!--   cat(sep="","1. Mutation: **", mutation, "**  \n") -->
<!--   target.drugs <- drugs %>% -->
<!--     filter(MutationSymbol == mutation) %>% -->
<!--     pull(Drug) %>% -->
<!--     unique() -->
<!--   target.regulators <- drugs %>% -->
<!--     filter(MutationSymbol == mutation) %>% -->
<!--     pull(RegulatorSymbol) %>% -->
<!--     unique() -->
<!--   target.regulons <- drugs %>% -->
<!--     filter(MutationSymbol == mutation) %>% -->
<!--     pull(Regulon_ID) %>% -->
<!--     unique() -->
<!--   target.list <- drugs %>% -->
<!--     filter(MutationSymbol == mutation) %>% -->
<!--     pull(TargetSymbol) %>% -->
<!--     unique() -->
<!--   rmsts <- drugs %>% -->
<!--     filter(MutationSymbol == mutation) %>% -->
<!--     pull(RMST_diff_UnderActiveMinusOverActiveMGMTMeth) -->
<!--   gbmtrials <- drugs %>% -->
<!--     filter(MutationSymbol == mutation) %>% -->
<!--     pull(DrugTrialPhaseGBM) %>% -->
<!--     unique() -->
<!--   mutation2regulators <- drugs %>% -->
<!--     filter(MutationSymbol == mutation) %>% -->
<!--     select(MutationRegulatorEdge) %>% -->
<!--     unique() %>% -->
<!--     mutate(`MutationRegulatorEdge` = if_else(MutationRegulatorEdge == "1", "activates", "suppresses")) %>% -->
<!--     pull() -->
<!--    anticancerPhaseIVs <- drugs %>% -->
<!--     filter(MutationSymbol == mutation) %>% -->
<!--     select(IfAntiCancerPhaseIV) %>% -->
<!--     unique() %>% -->
<!--     mutate(`IfAntiCancerPhaseIV` = if_else(IfAntiCancerPhaseIV == "AntiCancerPhaseIV", paste("**Phase IV** trial for some other cancer(s)"), paste(" **NOT** in Phase IV trials for any cancer."))) %>% -->
<!--      pull() -->
<!--    cat(sep = "","Evidence: Patient has **",mutation, "** mutation that **" ,mutation2regulators,"** regulator, ",target.regulators, " of **", length(target.regulons),  "** active risk-associated regulons containing the target(s), ", paste("**",paste(target.list, collapse=","),"**", sep=""), " of the drug **", paste(drug, collapse=","), "** with max RMST value of **", max(rmsts), "**.\n") -->
<!--   # cat(sep="", drug, " has **", paste(gbmtrials, collapse=","), "** clinical trials for GBM.  \n" ) -->
<!--   cat("  \n\n") -->
<!--   cat("  \n") -->
<!-- } -->
<!-- ``` -->
</div>
<div id="summary-table" class="section level3">
<h3>Summary Table</h3>
<div id="htmlwidget-e3cddfd3e29fdc7b18a6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e3cddfd3e29fdc7b18a6">{"x":{"style":"bootstrap4","filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"1\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.565217391304347\" data-max=\"1\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"2\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"logical\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;true&quot;,&quot;false&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"data":[["CHRNB1","GPD2,NDUFA4L2,NDUFC1","HRH1","HRH1","HRH1,SCN3A,SCN5A,SCN9A","KCNJ8","NR3C1","NR3C1","OXTR","RAF1","VEGFA","VEGFA","WEE1"],["ROCURONIUM","METFORMIN","DEXCHLORPHENIRAMINE, HISTAMINE, CYPROHEPTADINE","DEXCHLORPHENIRAMINE, HISTAMINE, CYPROHEPTADINE","ORPHENADRINE","DIAZOXIDE","DEXAMETHASONE, PREDNISONE, TRIAMCINOLONE, BUDESONIDE, HYDROCORTISONE, MIFEPRISTONE, PREDNISOLONE","DEXAMETHASONE, PREDNISONE, TRIAMCINOLONE, BUDESONIDE, HYDROCORTISONE, MIFEPRISTONE, PREDNISOLONE","CARBETOCIN, OXYTOCIN","SORAFENIB","BEVACIZUMAB, RANIBIZUMAB","BEVACIZUMAB, RANIBIZUMAB","ADAVOSERTIB"],[0.909090909090909,0.130434782608696,0.972972972972973,0.972972972972973,0.692307692307692,0,0,0,0.714285714285714,1,0.6875,0.6875,1],[1,0.565217391304348,1,1,0.692307692307692,1,1,1,1,1,1,1,1],["O: 3 U: 0 A: 4","O: 1 U: 0 A: 10","O: 15 U: 0 A: 16","O: 15 U: 0 A: 16","O: 15 U: 0 A: 22","O: 0 U: 0 A: 2","O: 0 U: 0 A: 1","O: 0 U: 0 A: 1","O: 3 U: 0 A: 5","O: 1 U: 0 A: 1","O: 4 U: 0 A: 5","O: 4 U: 0 A: 5","O: 2 U: 0 A: 2"],[0,2,0,1,0,0,2,0,0,2,2,0,1],[true,true,true,false,true,true,true,true,true,true,true,true,false],["","","","","","DNAH5","","","","","DNAH5","DNAH5",""]],"container":"<table class=\"table table-striped table-hover row-border order-column display\">\n  <thead>\n    <tr>\n      <th>Drug.Target<\/th>\n      <th>Drugs<\/th>\n      <th>Drug Const. Regulon Activity<\/th>\n      <th>Drug Const. Program Activity<\/th>\n      <th>Regulon Activity Summary<\/th>\n      <th>GBM Trials Phase<\/th>\n      <th>FDA Appr. Other<\/th>\n      <th>Upstream Mutation in Patient<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"headerCallback":"function(thead) {\n  $(thead).css('font-size', '80%');\n}","defaultStyle.fontSize":"10%","buttons":["pageLength","colvis",{"extend":"excel","title":"gbmSYGNAL Details"},{"extend":"pdf","title":"gbmSYGNAL Network Details","pageSize":"letter","orientation":"landscape","filename":"SYGNAL_Network_Details","customize":"function(doc){\n                                  doc.styles.tableHeader.color='white';\n                                  doc.defaultStyle.alignment = 'left';\n                                  doc.styles.tableHeader.alignment = 'left';\n                                  doc.pageMargins = [10,10,10,10];\n                                  doc.defaultStyle.fontSize = 7;\n                                  doc.styles.tableHeader.fontSize = 7;\n                                  doc.styles.title.fontSize = 9;\n                                  \n                                  }"}],"scrollX":true,"stateSave":false,"target":"row","dom":"Britp","columnDefs":[{"className":"dt-right","targets":[2,3,5]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-size':'80%'});\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'80%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'80%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'80%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'80%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'80%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'80%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'80%'});\n}"},"callback":"function(table) {\ntable.page(3).draw(false);\n}"},"evals":["options.headerCallback","options.buttons.3.customize","options.rowCallback","callback"],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.14.0 reshape2_1.4.4    plyr_1.8.6        jsonlite_1.7.2    gridExtra_2.3     here_1.0.1        DT_0.19           forcats_0.5.1     stringr_1.4.0     dplyr_1.0.5       purrr_0.3.4       readr_1.4.0       tidyr_1.1.3       tibble_3.1.0      ggplot2_3.3.5     tidyverse_1.3.1  

loaded via a namespace (and not attached):
 [1] httr_1.4.2        sass_0.3.1        modelr_0.1.8      bslib_0.2.4       shiny_1.6.0       assertthat_0.2.1  highr_0.9         cellranger_1.1.0  yaml_2.2.1        pillar_1.6.0      backports_1.2.1   glue_1.4.2        digest_0.6.27     promises_1.2.0.1  rvest_1.0.0       colorspace_2.0-0  htmltools_0.5.1.1 httpuv_1.5.5      pkgconfig_2.0.3   broom_0.7.6       haven_2.4.0       xtable_1.8-4      scales_1.1.1      whisker_0.4       later_1.1.0.1     git2r_0.28.0      generics_0.1.0    farver_2.1.0      ellipsis_0.3.1    withr_2.4.2       cli_3.0.1         magrittr_2.0.1    crayon_1.4.1      readxl_1.3.1      mime_0.10         evaluate_0.14     fs_1.5.0          fansi_0.4.2       xml2_1.3.2        tools_4.0.5       hms_1.0.0         lifecycle_1.0.0   munsell_0.5.0     reprex_2.0.0      compiler_4.0.5    jquerylib_0.1.3   rlang_0.4.10      grid_4.0.5        rstudioapi_0.13   htmlwidgets_1.5.3 crosstalk_1.1.1   labeling_0.4.2    rmarkdown_2.7     gtable_0.3.0      DBI_1.1.1         R6_2.5.0          lubridate_1.7.10  knitr_1.32        fastmap_1.1.0     utf8_1.2.1        workflowr_1.6.2   rprojroot_2.0.2   stringi_1.5.3     Rcpp_1.0.6        vctrs_0.3.7       dbplyr_2.1.1     
[67] tidyselect_1.1.0  xfun_0.22        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
