{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f0126e40-b83a-479d-b168-53b662b672c8",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Load libraries and model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "538c201f-1c2c-471a-a570-9e718976c21f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/serdarturkaslan/Documents/GitHub/GbmMINER/data/MINER_MicroLowessRNATMM.08.24.2020\n",
      "\n",
      "Loaded 505 Regulons.\n"
     ]
    }
   ],
   "source": [
    "#!/usr/bin/env python3\n",
    "# -*- coding: utf-8 -*-\n",
    "\n",
    "import os\n",
    "import sys\n",
    "import glob\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import miner as miner\n",
    "from scipy.stats import zscore\n",
    "import miner_py3_kk as miner\n",
    "from tqdm.notebook import tqdm, trange\n",
    "import time\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# change the working directory\n",
    "os.chdir('/Users/serdarturkaslan/Documents/GitHub/GbmMINER/data/MINER_MicroLowessRNATMM.08.24.2020/')\n",
    "\n",
    "# Path to the miner directory\n",
    "input_path = os.getcwd()\n",
    "print(input_path)\n",
    "\n",
    "# Load regulon Modules\n",
    "regulonModules = miner.read_json(os.path.join(input_path,\"regulons.json\"))\n",
    "regulonDfMicroGbmLatest = pd.read_csv(\"regulonDfGbmMicroRNASigCoxAndStatSig.csv\", header = 0)\n",
    "regulonDfMicroGbmLatest = list(regulonDfMicroGbmLatest['Regulon_ID'].drop_duplicates())\n",
    "regulonDfMicroGbmLatest = [str(i) for i in regulonDfMicroGbmLatest]\n",
    "regulonModules = dict((k, regulonModules[k]) for k in regulonDfMicroGbmLatest if k in regulonModules)\n",
    "\n",
    "# create regulon dictionary\n",
    "referenceDictionary = regulonModules\n",
    "print()\n",
    "print(\"Loaded \" + str(len(referenceDictionary)) + \" Regulons.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b6c62284-cc62-43df-8720-5e22cb7da226",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Load data folders\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "587262e6-7b48-43a4-b390-4e7db7e371f6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total Patients to process: 35\n"
     ]
    }
   ],
   "source": [
    "# Folders for patient data\n",
    "patientExpFiles = glob.glob('/Volumes/omics4tb2/SYGNAL/Swedish/results/RSEM_results/*')\n",
    "#patientDataFolders = list(filter(lambda x:'TL-' in x, patientDataFolders))\n",
    "print('Total Patients to process: %s' %(len(patientExpFiles)))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7947bf7e-949f-49b5-99c9-b40cb23d423d",
   "metadata": {},
   "source": [
    "# Load Protein coding genes for filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "3770de1f-f5f5-41b3-85b2-8bfc7433855a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Protein Coding Genes\n",
      "         Gene stable ID Transcript stable ID Gene name       Gene type  \\\n",
      "0       ENSG00000284967      ENST00000647178     FDFT1  protein_coding   \n",
      "1       ENSG00000284967      ENST00000647069     FDFT1  protein_coding   \n",
      "2       ENSG00000079459      ENST00000538689     FDFT1  protein_coding   \n",
      "3       ENSG00000079459      ENST00000615631     FDFT1  protein_coding   \n",
      "4       ENSG00000079459      ENST00000618539     FDFT1  protein_coding   \n",
      "...                 ...                  ...       ...             ...   \n",
      "251334  ENSG00000189171      ENST00000440685   S100A13  protein_coding   \n",
      "251335  ENSG00000189171      ENST00000476133   S100A13  protein_coding   \n",
      "251336  ENSG00000189171      ENST00000339556   S100A13  protein_coding   \n",
      "251337  ENSG00000189171      ENST00000392622   S100A13  protein_coding   \n",
      "251338  ENSG00000189171      ENST00000392623   S100A13  protein_coding   \n",
      "\n",
      "       Gene Synonym  \n",
      "0              DGPT  \n",
      "1              DGPT  \n",
      "2              DGPT  \n",
      "3              DGPT  \n",
      "4              DGPT  \n",
      "...             ...  \n",
      "251334          NaN  \n",
      "251335          NaN  \n",
      "251336          NaN  \n",
      "251337          NaN  \n",
      "251338          NaN  \n",
      "\n",
      "[251339 rows x 5 columns]\n",
      "\n",
      "Final list has total of 22492 protein coding genes.\n"
     ]
    }
   ],
   "source": [
    "# load protein coding genes list from ensembl for filtering\n",
    "ensemblProteinCodingGenes = pd.read_csv(os.path.join(input_path,\"homo_sapiens_protein_coding_export.txt\"), sep=\"\\t\", index_col=None, header = 0)\n",
    "print(\"Protein Coding Genes\")\n",
    "print(ensemblProteinCodingGenes)\n",
    "\n",
    "uniqueProteins = list(ensemblProteinCodingGenes['Gene stable ID'].drop_duplicates())\n",
    "print()\n",
    "print(\"Final list has total of \" + str(len(uniqueProteins)) + \" protein coding genes.\" )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc58fb42-3a4f-409d-b902-1b89e65c7d12",
   "metadata": {},
   "source": [
    "# Loop through each patient to calculate regulon activity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "3e7f9822-d4de-49cf-ac19-9271009b763a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b50afc384f224c3393124a8bb21b47c0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/35 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed: 01_SN505-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 36_SN804-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 14_SN579-Neurosphere-P1\n",
      "done!\n",
      "done!\n",
      "Processed: 15_SN598-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processed: 13_SN575-Neurosphere-P4\n",
      "done!\n",
      "done!\n",
      "Processed: 16_SN601-Neurosphere-p5\n",
      "done!\n",
      "done!\n",
      "Processed: 37_SN810-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processed: 39_SN820-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processed: 05_SN521-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 04_SN517-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 03_SN515-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 06_SN528-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 07_SN529-Neurosphere-P1\n",
      "done!\n",
      "done!\n",
      "Processed: 02_SN508-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 12_SN553-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 10_SN534-Neurosphere-P1\n",
      "done!\n",
      "done!\n",
      "Processed: 08_SN531-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 17_SN649-Adherent-P5\n",
      "done!\n",
      "done!\n",
      "Processed: 18_SN655-Neurosphere-P3\n",
      "done!\n",
      "done!\n",
      "Processed: 19_SN667-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 20_SN674-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 21_SN677-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processed: 22_SN678-Adherent-P3\n",
      "done!\n",
      "done!\n",
      "Processed: 23_SN683-Neurosphere-P3\n",
      "done!\n",
      "done!\n",
      "Processed: 26_SN720-Adherent-p2\n",
      "done!\n",
      "done!\n",
      "Processed: 27_SN735-Neurosphere-P9\n",
      "done!\n",
      "done!\n",
      "Processed: 28_SN747-Neurosphere-P4\n",
      "done!\n",
      "done!\n",
      "Processed: 29_SN753-Neurosphere-P3\n",
      "done!\n",
      "done!\n",
      "Processed: 30_SN756-Neurosphere-P6\n",
      "done!\n",
      "done!\n",
      "Processed: 31_SN758-Neurosphere-P6\n",
      "done!\n",
      "done!\n",
      "Processed: 32_SN767-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processed: 33_SN787-Neurosphere-P6\n",
      "done!\n",
      "done!\n",
      "Processed: 34_SN788-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processed: 35_SN799-Neurosphere-P3\n",
      "done!\n",
      "done!\n",
      "Processed: 38_SN812-Neurosphere-P6\n",
      "done!\n",
      "done!\n",
      "## Regulon Activity Summary\n",
      "                         Over  Under  Neutral\n",
      "01_SN505-Neurosphere-P2   184     54      267\n",
      "36_SN804-Neurosphere-P2   294      0      211\n",
      "14_SN579-Neurosphere-P1   209      2      294\n",
      "15_SN598-Neurosphere-P5   505      0        0\n",
      "13_SN575-Neurosphere-P4   173      2      330\n",
      "16_SN601-Neurosphere-p5   212      8      285\n",
      "37_SN810-Neurosphere-P5   229     15      261\n",
      "39_SN820-Neurosphere-P5   242     13      250\n",
      "05_SN521-Neurosphere-P2   197      2      306\n",
      "04_SN517-Neurosphere-P2   197     15      293\n",
      "03_SN515-Neurosphere-P2   233     17      255\n",
      "06_SN528-Neurosphere-P2   241     14      250\n",
      "07_SN529-Neurosphere-P1   505      0        0\n",
      "02_SN508-Neurosphere-P2    82     19      404\n",
      "12_SN553-Neurosphere-P2   186     19      300\n",
      "10_SN534-Neurosphere-P1   124     18      363\n",
      "08_SN531-Neurosphere-P2   173     16      316\n",
      "17_SN649-Adherent-P5      170     23      312\n",
      "18_SN655-Neurosphere-P3   128     22      355\n",
      "19_SN667-Neurosphere-P2   162     58      285\n",
      "20_SN674-Neurosphere-P2   213     15      277\n",
      "21_SN677-Neurosphere-P2   163     97      245\n",
      "22_SN678-Adherent-P3      292      3      210\n",
      "23_SN683-Neurosphere-P3   203     11      291\n",
      "26_SN720-Adherent-p2      256      4      245\n",
      "27_SN735-Neurosphere-P9   162     25      318\n",
      "28_SN747-Neurosphere-P4   190     14      301\n",
      "29_SN753-Neurosphere-P3   258      3      244\n",
      "30_SN756-Neurosphere-P6   199     18      288\n",
      "31_SN758-Neurosphere-P6   505      0        0\n",
      "32_SN767-Neurosphere-P5   188     17      300\n",
      "33_SN787-Neurosphere-P6   183     11      311\n",
      "34_SN788-Neurosphere-P5   131     18      356\n",
      "35_SN799-Neurosphere-P3   181     16      308\n",
      "38_SN812-Neurosphere-P6   170     15      320\n"
     ]
    }
   ],
   "source": [
    "# Patient analysis loop\n",
    "allSummary = pd.DataFrame()\n",
    "for expFile in tqdm(patientExpFiles):\n",
    "    time.sleep(0.01)\n",
    "    patientID = expFile.split('/')[-1] #get patientID\n",
    "    patientID = patientID.split('.genes.results')[0]\n",
    "    print(\"Processed: \" + patientID)\n",
    "    #patientDataFile = os.path.join(folder,\"RNA\",patientID + \".genes.results\") # get RNASeq results\n",
    "    \n",
    "    # Check if patient data file exists\n",
    "    if os.path.exists('%s' %(expFile)):\n",
    "        # create a new column with patient name\n",
    "        newColName = patientID + \"_zscore\"\n",
    "        \n",
    "        # Read expression data\n",
    "        rawExpressionData = pd.read_csv(expFile, sep=\"\\t\", index_col=None, header = 0)\n",
    "        \n",
    "        \n",
    "        # seperate ensembl gene ids and symbols\n",
    "        rawExpressionData[['GeneID','gene_symbol']] = pd.DataFrame(rawExpressionData).gene_id.str.split(\"_\",expand=True)\n",
    "        #print(rawExpressionData)\n",
    "        \n",
    "        # filter for protein coding genes\n",
    "        rawExpressionDataFilt = rawExpressionData[rawExpressionData.GeneID.isin(uniqueProteins)].copy()\n",
    "        #rawExpressionDataFilt = rawExpressionData.loc[rawExpressionData.GeneID in uniqueProteins]\n",
    "\n",
    "        #print(\"Filtered raw expression\")\n",
    "        #print(rawExpressionDataFilt)\n",
    "        \n",
    "        # zscore patients expression data\n",
    "        rawExpressionDataFilt[[newColName]] = rawExpressionDataFilt[['TPM']].apply(zscore)\n",
    "   \n",
    "        #expressionData = pd.read_csv(\"abundanceXCuresRNATPMTL1961DB85ZScoredLatest.csv\", sep=\",\", index_col=0, header = 0)\n",
    "        expressionData = rawExpressionDataFilt[[patientID + \"_zscore\"]]\n",
    "        expressionData.index = rawExpressionDataFilt['GeneID']\n",
    "        #print(expressionData)\n",
    "        \n",
    "        #print(\"Processed: \" + expFile)\n",
    "        \n",
    "        # calculate regulon activity\n",
    "        rr = miner.generateRegulonActivity(referenceDictionary,expressionData, p=0.05)\n",
    "        \n",
    "        # write regulon activity results to file\n",
    "        rr.to_csv(\"~/Documents/GitHub/GbmMINER/PD-GSC/Regulon_Activity_\" + patientID + \".csv\")\n",
    "   \n",
    "        # calculate regulon activity stats\n",
    "        overActiveRegulonCount = len(rr[rr[newColName] == 1])\n",
    "        underActiveRegulonCount = len(rr[rr[newColName] == -1])\n",
    "        neutralRegulonCount = len(rr[rr[newColName] == 0])\n",
    "        \n",
    "        resSummary = {\"Over\" : [overActiveRegulonCount],\n",
    "                              \"Under\" : [underActiveRegulonCount],\n",
    "                              \"Neutral\" : [neutralRegulonCount]}\n",
    "        resSummary = pd.DataFrame(resSummary)\n",
    "        resSummary = resSummary.rename(index={0: patientID})\n",
    "        \n",
    "        allSummary = allSummary.append(resSummary)\n",
    "        \n",
    "    else:\n",
    "        print(\"Data file doesnt exist: \" + expFile)\n",
    "\n",
    "print(\"## Regulon Activity Summary\")        \n",
    "print(allSummary)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcddfcf6-012a-45b6-b175-1708975e306c",
   "metadata": {},
   "source": [
    "# Loop through each patient to calculate Program activity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "e62203c5-833b-4527-94c7-ed705ffe33cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "520c06e4d17b4fee8a0ca6a0a1202ec0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/35 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing: 01_SN505-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 36_SN804-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 14_SN579-Neurosphere-P1\n",
      "done!\n",
      "done!\n",
      "Processing: 15_SN598-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processing: 13_SN575-Neurosphere-P4\n",
      "done!\n",
      "done!\n",
      "Processing: 16_SN601-Neurosphere-p5\n",
      "done!\n",
      "done!\n",
      "Processing: 37_SN810-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processing: 39_SN820-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processing: 05_SN521-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 04_SN517-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 03_SN515-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 06_SN528-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 07_SN529-Neurosphere-P1\n",
      "done!\n",
      "done!\n",
      "Processing: 02_SN508-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 12_SN553-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 10_SN534-Neurosphere-P1\n",
      "done!\n",
      "done!\n",
      "Processing: 08_SN531-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 17_SN649-Adherent-P5\n",
      "done!\n",
      "done!\n",
      "Processing: 18_SN655-Neurosphere-P3\n",
      "done!\n",
      "done!\n",
      "Processing: 19_SN667-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 20_SN674-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 21_SN677-Neurosphere-P2\n",
      "done!\n",
      "done!\n",
      "Processing: 22_SN678-Adherent-P3\n",
      "done!\n",
      "done!\n",
      "Processing: 23_SN683-Neurosphere-P3\n",
      "done!\n",
      "done!\n",
      "Processing: 26_SN720-Adherent-p2\n",
      "done!\n",
      "done!\n",
      "Processing: 27_SN735-Neurosphere-P9\n",
      "done!\n",
      "done!\n",
      "Processing: 28_SN747-Neurosphere-P4\n",
      "done!\n",
      "done!\n",
      "Processing: 29_SN753-Neurosphere-P3\n",
      "done!\n",
      "done!\n",
      "Processing: 30_SN756-Neurosphere-P6\n",
      "done!\n",
      "done!\n",
      "Processing: 31_SN758-Neurosphere-P6\n",
      "done!\n",
      "done!\n",
      "Processing: 32_SN767-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processing: 33_SN787-Neurosphere-P6\n",
      "done!\n",
      "done!\n",
      "Processing: 34_SN788-Neurosphere-P5\n",
      "done!\n",
      "done!\n",
      "Processing: 35_SN799-Neurosphere-P3\n",
      "done!\n",
      "done!\n",
      "Processing: 38_SN812-Neurosphere-P6\n",
      "done!\n",
      "done!\n",
      "## Program Activity Summary\n",
      "                         Over  Under  Neutral\n",
      "01_SN505-Neurosphere-P2    35      8       38\n",
      "36_SN804-Neurosphere-P2    43      0       38\n",
      "14_SN579-Neurosphere-P1    41      1       39\n",
      "15_SN598-Neurosphere-P5    81      0        0\n",
      "13_SN575-Neurosphere-P4    38      1       42\n",
      "16_SN601-Neurosphere-p5    35      2       44\n",
      "37_SN810-Neurosphere-P5    36      2       43\n",
      "39_SN820-Neurosphere-P5    38      3       40\n",
      "05_SN521-Neurosphere-P2    36      1       44\n",
      "04_SN517-Neurosphere-P2    40      3       38\n",
      "03_SN515-Neurosphere-P2    39      2       40\n",
      "06_SN528-Neurosphere-P2    35      2       44\n",
      "07_SN529-Neurosphere-P1    81      0        0\n",
      "02_SN508-Neurosphere-P2    25      2       54\n",
      "12_SN553-Neurosphere-P2    36      2       43\n",
      "10_SN534-Neurosphere-P1    28      2       51\n",
      "08_SN531-Neurosphere-P2    37      2       42\n",
      "17_SN649-Adherent-P5       37      2       42\n",
      "18_SN655-Neurosphere-P3    29      2       50\n",
      "19_SN667-Neurosphere-P2    34      9       38\n",
      "20_SN674-Neurosphere-P2    36      2       43\n",
      "21_SN677-Neurosphere-P2    34     17       30\n",
      "22_SN678-Adherent-P3       45      1       35\n",
      "23_SN683-Neurosphere-P3    40      2       39\n",
      "26_SN720-Adherent-p2       42      1       38\n",
      "27_SN735-Neurosphere-P9    34      2       45\n",
      "28_SN747-Neurosphere-P4    33      2       46\n",
      "29_SN753-Neurosphere-P3    40      1       40\n",
      "30_SN756-Neurosphere-P6    34      2       45\n",
      "31_SN758-Neurosphere-P6    81      0        0\n",
      "32_SN767-Neurosphere-P5    39      2       40\n",
      "33_SN787-Neurosphere-P6    30      2       49\n",
      "34_SN788-Neurosphere-P5    28      2       51\n",
      "35_SN799-Neurosphere-P3    39      2       40\n",
      "38_SN812-Neurosphere-P6    40      2       39\n"
     ]
    }
   ],
   "source": [
    "# Load Transcriptional programs\n",
    "transcriptional_programs = miner.read_json(os.path.join(input_path,\"transcriptional_programs.json\"))\n",
    "program_list = [transcriptional_programs[str(key)] for key in range(len(transcriptional_programs.keys()))]\n",
    "\n",
    "\n",
    "# Create dictionary of program genes\n",
    "# make dictionary of program keys with gene lists as elements\n",
    "pr_genes_expanded = {}\n",
    "for i in range(len(program_list)):\n",
    "    rgns = program_list[i]\n",
    "    genes = []\n",
    "    for r in rgns:\n",
    "        if r in regulonModules:\n",
    "            genes.append(regulonModules[r])\n",
    "    if len(genes) != 0:\n",
    "        genes = list(set(np.hstack(genes)))\n",
    "        pr_genes_expanded[i] = genes\n",
    "#print(regulonModules)        \n",
    "#print(pr_genes_expanded[1])\n",
    "        \n",
    "# Patient analysis loop\n",
    "allProgSummary = pd.DataFrame()\n",
    "for expFile in tqdm(patientExpFiles):\n",
    "    time.sleep(0.01)\n",
    "    patientID = expFile.split('/')[-1] #get patientID\n",
    "    patientID = patientID.split('.genes.results')[0]\n",
    "    print(\"Processing: \" + patientID)\n",
    "    \n",
    "    # Check if patient data file exists\n",
    "    if os.path.exists('%s' %(expFile)):\n",
    "        # create a new column with patient name\n",
    "        newColName = patientID + \"_zscore\"\n",
    "        \n",
    "        # Read expression data\n",
    "        rawExpressionData = pd.read_csv(expFile, sep=\"\\t\", index_col=None, header = 0)\n",
    "        \n",
    "        \n",
    "        # seperate ensembl gene ids and symbols\n",
    "        rawExpressionData[['GeneID','gene_symbol']] = pd.DataFrame(rawExpressionData).gene_id.str.split(\"_\",expand=True)\n",
    "        #print(rawExpressionData)\n",
    "        \n",
    "        # filter for protein coding genes\n",
    "        rawExpressionDataFilt = rawExpressionData[rawExpressionData.GeneID.isin(uniqueProteins)].copy()\n",
    "        #rawExpressionDataFilt = rawExpressionData\n",
    "\n",
    "        #print(\"Filtered raw expression\")\n",
    "        #print(rawExpressionDataFilt)\n",
    "        \n",
    "        # zscore patients expression data\n",
    "        rawExpressionDataFilt[[newColName]] = rawExpressionDataFilt[['TPM']].apply(zscore)\n",
    "   \n",
    "        #expressionData = pd.read_csv(\"abundanceXCuresRNATPMTL1961DB85ZScoredLatest.csv\", sep=\",\", index_col=0, header = 0)\n",
    "        expressionData = rawExpressionDataFilt[[patientID + \"_zscore\"]]\n",
    "        expressionData.index = rawExpressionDataFilt['GeneID']\n",
    "        #print(expressionData)\n",
    "        \n",
    "        #print(\"Processed: \" + expFile)\n",
    "           \n",
    "        # Calculate program activity\n",
    "        #pp = miner.generateProgramActivity(transcriptional_programs = transcriptional_programs, regulonModules=referenceDictionary, expressionData=expressionData, p=0.05)   \n",
    "        \n",
    "        # select reference dictionary for downstream analysis (pr_genes, revisedClusters, coexpressionModules, or regulonModules)\n",
    "        referenceDictionaryPr = pr_genes_expanded\n",
    "        \n",
    "        # create a background matrix used for statistical hypothesis testing\n",
    "        bkgd = miner.backgroundDf(expressionData)\n",
    "        #print(bkgd)\n",
    "        # for each cluster, give samples that show high coherent cluster activity\n",
    "        overExpressedMembers = miner.biclusterMembershipDictionary(referenceDictionaryPr,bkgd,label=2,p=0.05)\n",
    "        \n",
    "        # for each clus|ter, give samples that show low coherent cluster activity\n",
    "        underExpressedMembers = miner.biclusterMembershipDictionary(referenceDictionaryPr,bkgd,label=0,p=0.05)\n",
    "        \n",
    "        # convert overExpressedMembers dictionary to binary matrix\n",
    "        overExpressedMembersMatrix = miner.membershipToIncidence(overExpressedMembers,expressionData)\n",
    "        #overExpressedMembersMatrix.index = np.array(overExpressedMembersMatrix.index).astype(str)\n",
    "        #print(overExpressedMembersMatrix)\n",
    "        # convert underExpressedMembers dictionary to binary matrix\n",
    "        underExpressedMembersMatrix = miner.membershipToIncidence(underExpressedMembers,expressionData)\n",
    "        #underExpressedMembersMatrix.index = np.array(underExpressedMembersMatrix.index).astype(str)\n",
    "        #print(underExpressedMembersMatrix)\n",
    "        # Create program matrix with values of {-1,0,1}\n",
    "        pp = overExpressedMembersMatrix-underExpressedMembersMatrix\n",
    "        \n",
    "        ## Histogram plot\n",
    "        #plt.hist(pp, bins='auto')\n",
    "        #plt.title(\"Program activity for \" + patientID)\n",
    "        #plt.show()\n",
    "        #print(pp)\n",
    "\n",
    "        # write Program activity results to file\n",
    "        pp.to_csv(\"~/Documents/GitHub/GbmMINER/PD-GSC/Program_Activity_\" + patientID + \".csv\")\n",
    "        \n",
    "        # calculate Program activity stats\n",
    "        overActiveProgramCount = len(pp[pp[newColName] == 1])\n",
    "        underActiveProgramCount = len(pp[pp[newColName] == -1])\n",
    "        neutralProgramCount = len(pp[pp[newColName] == 0])\n",
    "        \n",
    "        resProgSummary = {\n",
    "            \"Over\" : [overActiveProgramCount],\n",
    "            \"Under\" : [underActiveProgramCount],\n",
    "            \"Neutral\" : [neutralProgramCount]\n",
    "        }\n",
    "        \n",
    "        resProgSummary = pd.DataFrame(resProgSummary)\n",
    "        resProgSummary = resProgSummary.rename(index={0: patientID})\n",
    "        \n",
    "        allProgSummary = allProgSummary.append(resProgSummary)\n",
    "        \n",
    "        \n",
    "    else:\n",
    "        print(\"Data file doesnt exist: \" + expFile)\n",
    "\n",
    "print(\"## Program Activity Summary\")\n",
    "print(allProgSummary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "fee54d34-1824-4ad6-9edb-8630636bb6b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([ 2.,  0.,  0.,  0., 51.,  0.,  0., 28.]),\n",
       " array([-1.  , -0.75, -0.5 , -0.25,  0.  ,  0.25,  0.5 ,  0.75,  1.  ]),\n",
       " <BarContainer object of 8 artists>)"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXAAAAD4CAYAAAD1jb0+AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8rg+JYAAAACXBIWXMAAAsTAAALEwEAmpwYAAAPQ0lEQVR4nO3dfYxld13H8feHXdqqiN2l47q0hGlDBZsYWjKpVYxIy0OhprvGWpeILrpmBdFg0Mgi/6jR2PqHVaMJbnhan/rgYtOVirhs2xATWphKgT5YdltK7LrtDtAixFhp+frHPYPX2Tt778zce2d/8n4lk3vO7/zOOd/9nbufOXPuOTOpKiRJ7XnWehcgSVodA1ySGmWAS1KjDHBJapQBLkmN2jjNnZ111lk1Ozs7zV1KUvPuvvvuL1bVzNL2qQb47Ows8/Pz09ylJDUvyRcGtXsJRZIaZYBLUqMMcElqlAEuSY0ywCWpUQa4JDXKAJekRhngktQoA1ySGjXVJzGlU9XsnlvXu4SBHrnmivUuQaewkQI8ySPAV4FngKerai7JZuBGYBZ4BLi6qp6YTJmSpKVWcgnllVV1YVXNdfN7gENVdT5wqJuXJE3JWq6BbwP2ddP7gO1rrkaSNLJRA7yAf0pyd5LdXduWqjrWTT8GbBm0YpLdSeaTzC8sLKyxXEnSolE/xPzhqjqa5LuBg0n+tX9hVVWSgX/evqr2AnsB5ubmBvaRJK3cSGfgVXW0ez0O3AxcDDyeZCtA93p8UkVKkk40NMCTfEeS71ycBl4D3AscAHZ23XYCt0yqSEnSiUa5hLIFuDnJYv+/qap/TPJJ4KYku4AvAFdPrkxJ0lJDA7yqHgZeOqD9S8BlkyhKkjScj9JLUqMMcElqlAEuSY0ywCWpUQa4JDXKAJekRhngktQoA1ySGmWAS1KjDHBJapQBLkmNMsAlqVEGuCQ1ygCXpEYZ4JLUKANckhplgEtSowxwSWqUAS5JjTLAJalRBrgkNcoAl6RGGeCS1CgDXJIaZYBLUqMMcElqlAEuSY0ywCWpUQa4JDXKAJekRhngktSokQM8yYYkn0ryoW7+3CR3JTmS5MYkp02uTEnSUis5A38b8EDf/LXAdVX1IuAJYNc4C5MkndxIAZ7kHOAK4D3dfIBLgf1dl33A9gnUJ0laxqhn4H8E/AbwjW7+ecCTVfV0N/8ocPagFZPsTjKfZH5hYWEttUqS+gwN8CQ/BhyvqrtXs4Oq2ltVc1U1NzMzs5pNSJIG2DhCn5cDVyZ5PXAG8Fzgj4Ezk2zszsLPAY5OrkxJ0lJDz8Cr6p1VdU5VzQI7gNuq6qeB24Grum47gVsmVqUk6QRruQ/8HcDbkxyhd038veMpSZI0ilEuoXxTVd0B3NFNPwxcPP6SJEmj8ElMSWqUAS5JjTLAJalRBrgkNcoAl6RGGeCS1CgDXJIaZYBLUqMMcElqlAEuSY0ywCWpUQa4JDXKAJekRhngktQoA1ySGmWAS1KjDHBJapQBLkmNMsAlqVEGuCQ1ygCXpEYZ4JLUKANckhplgEtSowxwSWqUAS5JjTLAJalRBrgkNcoAl6RGGeCS1CgDXJIaZYBLUqOGBniSM5J8Ismnk9yX5Le79nOT3JXkSJIbk5w2+XIlSYtGOQN/Cri0ql4KXAhcnuQS4Frguqp6EfAEsGtiVUqSTjA0wKvna93ss7uvAi4F9nft+4DtkyhQkjTYSNfAk2xIcg9wHDgIPAQ8WVVPd10eBc5eZt3dSeaTzC8sLIyhZEkSjBjgVfVMVV0InANcDLxk1B1U1d6qmququZmZmdVVKUk6wYruQqmqJ4HbgR8EzkyysVt0DnB0vKVJkk5mlLtQZpKc2U1/G/Bq4AF6QX5V120ncMuEapQkDbBxeBe2AvuSbKAX+DdV1YeS3A/ckOR3gU8B751gnZKkJYYGeFV9BrhoQPvD9K6HS5LWgU9iSlKjDHBJatQo18AlqQmze25d7xIGeuSaKyayXc/AJalRBrgkNcoAl6RGGeCS1CgDXJIaZYBLUqMMcElqlAEuSY0ywCWpUQa4JDXKAJekRhngktQoA1ySGmWAS1KjDHBJapQBLkmNMsAlqVEGuCQ1ygCXpEYZ4JLUKANckhplgEtSowxwSWqUAS5JjTLAJalRBrgkNcoAl6RGGeCS1KihAZ7kBUluT3J/kvuSvK1r35zkYJLD3eumyZcrSVo0yhn408CvVdUFwCXAW5NcAOwBDlXV+cChbl6SNCVDA7yqjlXVv3TTXwUeAM4GtgH7um77gO0TqlGSNMCKroEnmQUuAu4CtlTVsW7RY8CW8ZYmSTqZkQM8yXOADwK/WlX/0b+sqgqoZdbbnWQ+yfzCwsKaipUk/a+RAjzJs+mF919X1d91zY8n2dot3wocH7RuVe2tqrmqmpuZmRlHzZIkRrsLJcB7gQeq6g/7Fh0AdnbTO4Fbxl+eJGk5G0fo83LgZ4DPJrmna/tN4BrgpiS7gC8AV0+kQknSQEMDvKr+Gcgyiy8bbzmSpFH5JKYkNcoAl6RGGeCS1CgDXJIaZYBLUqMMcElqlAEuSY0ywCWpUQa4JDXKAJekRhngktQoA1ySGmWAS1KjDHBJapQBLkmNMsAlqVEGuCQ1ygCXpEYZ4JLUKANckhplgEtSowxwSWqUAS5JjTLAJalRBrgkNcoAl6RGGeCS1CgDXJIaZYBLUqMMcElqlAEuSY0ywCWpUUMDPMn7khxPcm9f2+YkB5Mc7l43TbZMSdJSo5yBfwC4fEnbHuBQVZ0PHOrmJUlTNDTAq+pjwJeXNG8D9nXT+4Dt4y1LkjTMaq+Bb6mqY930Y8CW5Tom2Z1kPsn8wsLCKncnSVpqzR9iVlUBdZLle6tqrqrmZmZm1ro7SVJntQH+eJKtAN3r8fGVJEkaxWoD/ACws5veCdwynnIkSaMa5TbC64GPAy9O8miSXcA1wKuTHAZe1c1LkqZo47AOVfWGZRZdNuZaJEkr4JOYktQoA1ySGmWAS1KjDHBJapQBLkmNMsAlqVEGuCQ1ygCXpEYZ4JLUKANckhplgEtSowxwSWqUAS5JjTLAJalRBrgkNcoAl6RGGeCS1CgDXJIaZYBLUqMMcElqlAEuSY0ywCWpUQa4JDXKAJekRhngktQoA1ySGmWAS1KjNq53AaOa3XPrepcw0CPXXLHeJUj6FuUZuCQ1ygCXpEYZ4JLUKANckhq1pgBPcnmSB5McSbJnXEVJkoZbdYAn2QD8GfA64ALgDUkuGFdhkqSTW8sZ+MXAkap6uKr+G7gB2DaesiRJw6zlPvCzgX/rm38U+IGlnZLsBnZ3s19L8uAq93cW8MVVrjsxufbUrItTdLywrhXx/bVip2RdYziOLxzUOPEHeapqL7B3rdtJMl9Vc2Moaaysa2Wsa2Wsa2W+1epayyWUo8AL+ubP6dokSVOwlgD/JHB+knOTnAbsAA6MpyxJ0jCrvoRSVU8n+WXgI8AG4H1Vdd/YKjvRmi/DTIh1rYx1rYx1rcy3VF2pqklsV5I0YT6JKUmNMsAlqVGnVIAn+ckk9yX5RpJlb7lZ7hH+7gPVu7r2G7sPV8dR1+YkB5Mc7l43DejzyiT39H39V5Lt3bIPJPl837ILp1VX1++Zvn0f6Gtfz/G6MMnHu+P9mSQ/1bdsrOM17Fc+JDm9+/cf6cZjtm/ZO7v2B5O8di11rKKutye5vxufQ0le2Lds4DGdUl1vSrLQt/9f6Fu2szvuh5PsnHJd1/XV9LkkT/Ytm8h4JXlfkuNJ7l1meZL8SVfzZ5K8rG/Z2seqqk6ZL+D7gBcDdwBzy/TZADwEnAecBnwauKBbdhOwo5t+N/CWMdX1B8CebnoPcO2Q/puBLwPf3s1/ALhqAuM1Ul3A15ZpX7fxAr4XOL+bfj5wDDhz3ON1svdLX59fAt7dTe8AbuymL+j6nw6c221nwxTremXfe+gti3Wd7JhOqa43AX86YN3NwMPd66ZuetO06lrS/1fo3Vgx6fH6EeBlwL3LLH898GEgwCXAXeMcq1PqDLyqHqiqYU9qDnyEP0mAS4H9Xb99wPYxlbat296o270K+HBV/eeY9r+cldb1Tes9XlX1uao63E3/O3AcmBnT/vuN8isf+uvdD1zWjc824IaqeqqqPg8c6bY3lbqq6va+99Cd9J61mLS1/IqM1wIHq+rLVfUEcBC4fJ3qegNw/Zj2vayq+hi9k7XlbAP+onruBM5MspUxjdUpFeAjGvQI/9nA84Anq+rpJe3jsKWqjnXTjwFbhvTfwYlvnt/rfoS6LsnpU67rjCTzSe5cvKzDKTReSS6md1b1UF/zuMZruffLwD7deHyF3viMsu4k6+q3i96Z3KJBx3Sadf1Ed3z2J1l8oO+UGK/uUtO5wG19zZMar2GWq3ssYzX1v4mZ5KPA9wxY9K6qumXa9Sw6WV39M1VVSZa997L77vr99O6PX/ROekF2Gr37Qd8B/M4U63phVR1Nch5wW5LP0gupVRvzeP0lsLOqvtE1r3q8/j9K8kZgDnhFX/MJx7SqHhq8hbH7e+D6qnoqyS/S++nl0intexQ7gP1V9Uxf23qO18RMPcCr6lVr3MRyj/B/id6PJxu7s6gVPdp/srqSPJ5ka1Ud6wLn+Ek2dTVwc1V9vW/bi2ejTyV5P/Dr06yrqo52rw8nuQO4CPgg6zxeSZ4L3Ervm/edfdte9XgNMMqvfFjs82iSjcB30Xs/TfLXRYy07SSvovdN8RVV9dRi+zLHdByBNLSuqvpS3+x76H3msbjujy5Z944x1DRSXX12AG/tb5jgeA2zXN1jGasWL6EMfIS/ep8M3E7v+jPATmBcZ/QHuu2Nst0Trr11IbZ43Xk7MPAT60nUlWTT4iWIJGcBLwfuX+/x6o7dzfSuD+5fsmyc4zXKr3zor/cq4LZufA4AO9K7S+Vc4HzgE2uoZUV1JbkI+HPgyqo63tc+8JhOsa6tfbNXAg900x8BXtPVtwl4Df/3J9GJ1tXV9hJ6Hwp+vK9tkuM1zAHgZ7u7US4BvtKdoIxnrCbxyexqv4Afp3ct6CngceAjXfvzgX/o6/d64HP0voO+q6/9PHr/wY4AfwucPqa6ngccAg4DHwU2d+1zwHv6+s3S+876rCXr3wZ8ll4Q/RXwnGnVBfxQt+9Pd6+7ToXxAt4IfB24p+/rwkmM16D3C71LMld202d0//4j3Xic17fuu7r1HgReN+b3+7C6Ptr9P1gcnwPDjumU6vp94L5u/7cDL+lb9+e7cTwC/Nw06+rmfwu4Zsl6Exsveidrx7r38qP0Pqt4M/Dmbnno/eGbh7p9z/Wtu+ax8lF6SWpUi5dQJEkY4JLULANckhplgEtSowxwSWqUAS5JjTLAJalR/wPN91B+EgItzgAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.hist(pp, bins='auto')\n",
    "\n",
    "pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "7e3260c2-7eaf-4b17-9ffc-33bad119c01c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([6547.,    0.,    0.,    0.,    0., 6543.,    0.,    0.,    0.,\n",
       "        6548.]),\n",
       " array([-1. , -0.8, -0.6, -0.4, -0.2,  0. ,  0.2,  0.4,  0.6,  0.8,  1. ]),\n",
       " <BarContainer object of 10 artists>)"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAX0AAAD4CAYAAAAAczaOAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8rg+JYAAAACXBIWXMAAAsTAAALEwEAmpwYAAATqUlEQVR4nO3dYbBc5X3f8e/PKOA2aSzJqCqRGEueqHHIdAzMHSB1J4khFYJ0LGWKiTxJUag6alKaSaftNFC/oMVhavqiNEwTUo1RLNzUmCj1oCYkRBYwmc4EjKgxNhCsCw6DVEA3SNC6jInB/77Y57prca/uXmnvCvJ8PzN39pznPOec/3l2+e3R2bNLqgpJUh/edboLkCRNjqEvSR0x9CWpI4a+JHXE0Jekjiw73QWcyNlnn13r1q073WVI0jvKo48++udVtWquZW/r0F+3bh0HDhw43WVI0jtKkufmW+blHUnqiKEvSR0x9CWpI4a+JHXE0Jekjhj6ktQRQ1+SOmLoS1JHDH1J6sjb+hu5p2rd9b9/Wvb7Z5/8qdOyX02Wr6+//E7XcwxL9zx7pi9JHTH0Jakjhr4kdcTQl6SOGPqS1BFDX5I6YuhLUkcMfUnqiKEvSR0x9CWpI4a+JHXE0Jekjhj6ktSRkUI/yfIke5L8aZKnkvxokpVJ9iU52B5XtL5JcluS6SSPJ7lwaDvbWv+DSbYt1UFJkuY26pn+rwF/WFUfAD4IPAVcD+yvqg3A/jYPcAWwof3tAG4HSLISuBG4GLgIuHH2jUKSNBkLhn6S9wA/BtwBUFV/UVWvAJuB3a3bbmBLm94M3FkDDwHLk5wDXA7sq6qjVXUM2AdsGuOxSJIWMMqZ/npgBvitJF9K8qkk3wusrqoXWp8XgdVteg3w/ND6h1rbfO3fJcmOJAeSHJiZmVnc0UiSTmiU0F8GXAjcXlUXAP+X/38pB4CqKqDGUVBV7ayqqaqaWrVq1Tg2KUlqRgn9Q8Chqnq4ze9h8CbwUrtsQ3s80pYfBs4dWn9ta5uvXZI0IQuGflW9CDyf5Ida02XAk8BeYPYOnG3APW16L3BNu4vnEuDVdhnoPmBjkhXtA9yNrU2SNCGj/o/Rfwn47SRnAs8C1zJ4w7g7yXbgOeDq1vde4EpgGnit9aWqjib5BPBI63dTVR0dy1FIkkYyUuhX1WPA1ByLLpujbwHXzbOdXcCuRdQnSRojv5ErSR0x9CWpI4a+JHXE0Jekjhj6ktQRQ1+SOmLoS1JHDH1J6oihL0kdMfQlqSOGviR1xNCXpI4Y+pLUEUNfkjpi6EtSRwx9SeqIoS9JHTH0Jakjhr4kdcTQl6SOGPqS1BFDX5I6YuhLUkdGCv0kf5bkK0keS3Kgta1Msi/Jwfa4orUnyW1JppM8nuTCoe1sa/0PJtm2NIckSZrPYs70P1xV51fVVJu/HthfVRuA/W0e4ApgQ/vbAdwOgzcJ4EbgYuAi4MbZNwpJ0mScyuWdzcDuNr0b2DLUfmcNPAQsT3IOcDmwr6qOVtUxYB+w6RT2L0lapFFDv4A/SvJokh2tbXVVvdCmXwRWt+k1wPND6x5qbfO1f5ckO5IcSHJgZmZmxPIkSaNYNmK/v1NVh5P8dWBfkj8dXlhVlaTGUVBV7QR2AkxNTY1lm5KkgZHO9KvqcHs8AnyewTX5l9plG9rjkdb9MHDu0OprW9t87ZKkCVkw9JN8b5K/NjsNbAS+CuwFZu/A2Qbc06b3Ate0u3guAV5tl4HuAzYmWdE+wN3Y2iRJEzLK5Z3VwOeTzPb/r1X1h0keAe5Osh14Dri69b8XuBKYBl4DrgWoqqNJPgE80vrdVFVHx3YkkqQFLRj6VfUs8ME52l8GLpujvYDr5tnWLmDX4suUJI2D38iVpI4Y+pLUEUNfkjpi6EtSRwx9SeqIoS9JHTH0Jakjhr4kdcTQl6SOGPqS1BFDX5I6YuhLUkcMfUnqiKEvSR0x9CWpI4a+JHXE0Jekjhj6ktQRQ1+SOmLoS1JHDH1J6oihL0kdGTn0k5yR5EtJfq/Nr0/ycJLpJJ9LcmZrP6vNT7fl64a2cUNrfzrJ5WM/GknSCS3mTP+XgaeG5m8Bbq2qHwSOAdtb+3bgWGu/tfUjyXnAVuBHgE3AbyQ549TKlyQtxkihn2Qt8FPAp9p8gEuBPa3LbmBLm97c5mnLL2v9NwN3VdXrVfV1YBq4aAzHIEka0ahn+v8R+FfAt9v8e4FXquqNNn8IWNOm1wDPA7Tlr7b+32mfY53vSLIjyYEkB2ZmZkY/EknSghYM/SR/DzhSVY9OoB6qamdVTVXV1KpVqyaxS0nqxrIR+nwI+EiSK4F3A98P/BqwPMmydja/Fjjc+h8GzgUOJVkGvAd4eah91vA6kqQJWPBMv6puqKq1VbWOwQex91fVzwIPAFe1btuAe9r03jZPW35/VVVr39ru7lkPbAC+OLYjkSQtaJQz/fn8CnBXkl8FvgTc0drvAD6TZBo4yuCNgqp6IsndwJPAG8B1VfXmKexfkrRIiwr9qnoQeLBNP8scd99U1TeBj86z/s3AzYstUpI0Hn4jV5I6YuhLUkcMfUnqiKEvSR0x9CWpI4a+JHXE0Jekjhj6ktQRQ1+SOmLoS1JHDH1J6oihL0kdMfQlqSOGviR1xNCXpI4Y+pLUEUNfkjpi6EtSRwx9SeqIoS9JHTH0Jakjhr4kdcTQl6SOLBj6Sd6d5ItJvpzkiST/trWvT/Jwkukkn0tyZms/q81Pt+XrhrZ1Q2t/OsnlS3ZUkqQ5jXKm/zpwaVV9EDgf2JTkEuAW4Naq+kHgGLC99d8OHGvtt7Z+JDkP2Ar8CLAJ+I0kZ4zxWCRJC1gw9GvgG232e9pfAZcCe1r7bmBLm97c5mnLL0uS1n5XVb1eVV8HpoGLxnEQkqTRjHRNP8kZSR4DjgD7gGeAV6rqjdblELCmTa8Bngdoy18F3jvcPsc6w/vakeRAkgMzMzOLPiBJ0vxGCv2qerOqzgfWMjg7/8BSFVRVO6tqqqqmVq1atVS7kaQuLerunap6BXgA+FFgeZJlbdFa4HCbPgycC9CWvwd4ebh9jnUkSRMwyt07q5Isb9N/Bfi7wFMMwv+q1m0bcE+b3tvmacvvr6pq7Vvb3T3rgQ3AF8d0HJKkESxbuAvnALvbnTbvAu6uqt9L8iRwV5JfBb4E3NH63wF8Jsk0cJTBHTtU1RNJ7gaeBN4ArquqN8d7OJKkE1kw9KvqceCCOdqfZY67b6rqm8BH59nWzcDNiy9TkjQOfiNXkjpi6EtSRwx9SeqIoS9JHTH0Jakjhr4kdcTQl6SOGPqS1BFDX5I6YuhLUkcMfUnqiKEvSR0x9CWpI4a+JHXE0Jekjhj6ktQRQ1+SOmLoS1JHDH1J6oihL0kdMfQlqSOGviR1ZMHQT3JukgeSPJnkiSS/3NpXJtmX5GB7XNHak+S2JNNJHk9y4dC2trX+B5NsW7rDkiTNZZQz/TeAf1FV5wGXANclOQ+4HthfVRuA/W0e4ApgQ/vbAdwOgzcJ4EbgYuAi4MbZNwpJ0mQsGPpV9UJV/c82/X+Ap4A1wGZgd+u2G9jSpjcDd9bAQ8DyJOcAlwP7qupoVR0D9gGbxnkwkqQTW9Q1/STrgAuAh4HVVfVCW/QisLpNrwGeH1rtUGubr12SNCEjh36S7wN+F/hnVfW/h5dVVQE1joKS7EhyIMmBmZmZcWxSktSMFPpJvodB4P92Vf231vxSu2xDezzS2g8D5w6tvra1zdf+XapqZ1VNVdXUqlWrFnMskqQFjHL3ToA7gKeq6j8MLdoLzN6Bsw24Z6j9mnYXzyXAq+0y0H3AxiQr2ge4G1ubJGlClo3Q50PAPwC+kuSx1vavgU8CdyfZDjwHXN2W3QtcCUwDrwHXAlTV0SSfAB5p/W6qqqPjOAhJ0mgWDP2q+h9A5ll82Rz9C7hunm3tAnYtpkBJ0vj4jVxJ6oihL0kdMfQlqSOGviR1xNCXpI4Y+pLUEUNfkjpi6EtSRwx9SeqIoS9JHTH0Jakjhr4kdcTQl6SOGPqS1BFDX5I6YuhLUkcMfUnqiKEvSR0x9CWpI4a+JHXE0Jekjhj6ktQRQ1+SOrJg6CfZleRIkq8Ota1Msi/Jwfa4orUnyW1JppM8nuTCoXW2tf4Hk2xbmsORJJ3IKGf6nwY2Hdd2PbC/qjYA+9s8wBXAhva3A7gdBm8SwI3AxcBFwI2zbxSSpMlZMPSr6o+Bo8c1bwZ2t+ndwJah9jtr4CFgeZJzgMuBfVV1tKqOAft46xuJJGmJnew1/dVV9UKbfhFY3abXAM8P9TvU2uZrf4skO5IcSHJgZmbmJMuTJM3llD/IraoCagy1zG5vZ1VNVdXUqlWrxrVZSRInH/ovtcs2tMcjrf0wcO5Qv7Wtbb52SdIEnWzo7wVm78DZBtwz1H5Nu4vnEuDVdhnoPmBjkhXtA9yNrU2SNEHLFuqQ5LPATwBnJznE4C6cTwJ3J9kOPAdc3brfC1wJTAOvAdcCVNXRJJ8AHmn9bqqq4z8cliQtsQVDv6o+Ns+iy+boW8B182xnF7BrUdVJksbKb+RKUkcMfUnqiKEvSR0x9CWpI4a+JHXE0Jekjhj6ktQRQ1+SOmLoS1JHDH1J6oihL0kdMfQlqSOGviR1xNCXpI4Y+pLUEUNfkjpi6EtSRwx9SeqIoS9JHTH0Jakjhr4kdcTQl6SOGPqS1JGJh36STUmeTjKd5PpJ71+SejbR0E9yBvDrwBXAecDHkpw3yRokqWeTPtO/CJiuqmer6i+Au4DNE65Bkrq1bML7WwM8PzR/CLh4uEOSHcCONvuNJE+fwv7OBv78FNY/KbllwS6npa4RWNfi+PpaHOtahNxySnW9b74Fkw79BVXVTmDnOLaV5EBVTY1jW+NkXYtjXYtjXYvTW12TvrxzGDh3aH5ta5MkTcCkQ/8RYEOS9UnOBLYCeydcgyR1a6KXd6rqjST/FLgPOAPYVVVPLOEux3KZaAlY1+JY1+JY1+J0VVeqaim2K0l6G/IbuZLUEUNfkjryjg79JB9N8kSSbyeZ99am+X76oX2g/HBr/1z7cHkcda1Msi/Jwfa4Yo4+H07y2NDfN5Nsacs+neTrQ8vOn1Rdrd+bQ/veO9R+Osfr/CR/0p7vx5P8zNCysY7XQj8VkuSsdvzTbTzWDS27obU/neTyU6njJOr650mebOOzP8n7hpbN+ZxOqK6fTzIztP9/NLRsW3veDybZNuG6bh2q6WtJXhlatpTjtSvJkSRfnWd5ktzW6n48yYVDy059vKrqHfsH/DDwQ8CDwNQ8fc4AngHeD5wJfBk4ry27G9japn8T+MUx1fXvgevb9PXALQv0XwkcBf5qm/80cNUSjNdIdQHfmKf9tI0X8DeBDW36B4AXgOXjHq8TvV6G+vwT4Dfb9Fbgc236vNb/LGB9284ZE6zrw0OvoV+cretEz+mE6vp54D/Nse5K4Nn2uKJNr5hUXcf1/yUGN5Ys6Xi1bf8YcCHw1XmWXwn8ARDgEuDhcY7XO/pMv6qeqqqFvrE7508/JAlwKbCn9dsNbBlTaZvb9kbd7lXAH1TVa2Pa/3wWW9d3nO7xqqqvVdXBNv2/gCPAqjHtf9goPxUyXO8e4LI2PpuBu6rq9ar6OjDdtjeRuqrqgaHX0EMMvgez1E7lp1UuB/ZV1dGqOgbsAzadpro+Bnx2TPs+oar6YwYnefPZDNxZAw8By5Ocw5jG6x0d+iOa66cf1gDvBV6pqjeOax+H1VX1Qpt+EVi9QP+tvPUFd3P7p92tSc6acF3vTnIgyUOzl5x4G41XkosYnL09M9Q8rvGa7/UyZ582Hq8yGJ9R1l3KuoZtZ3C2OGuu53SSdf399vzsSTL7Bc23xXi1y2DrgfuHmpdqvEYxX+1jGa+33c8wHC/JF4C/Mceij1fVPZOuZ9aJ6hqeqapKMu99se0d/G8x+O7CrBsYhN+ZDO7V/RXgpgnW9b6qOpzk/cD9Sb7CINhO2pjH6zPAtqr6dms+6fH6yyjJzwFTwI8PNb/lOa2qZ+bewtj9d+CzVfV6kn/M4F9Jl05o36PYCuypqjeH2k7neC2pt33oV9VPnuIm5vvph5cZ/LNpWTtbW9RPQpyoriQvJTmnql5oIXXkBJu6Gvh8VX1raNuzZ72vJ/kt4F9Osq6qOtwen03yIHAB8Luc5vFK8v3A7zN4w39oaNsnPV5zGOWnQmb7HEqyDHgPg9fTUv7MyEjbTvKTDN5If7yqXp9tn+c5HUeILVhXVb08NPspBp/hzK77E8et++AYahqpriFbgeuGG5ZwvEYxX+1jGa8eLu/M+dMPNfhk5AEG19MBtgHj+pfD3ra9Ubb7lmuJLfhmr6NvAeb8lH8p6kqyYvbySJKzgQ8BT57u8WrP3ecZXOvcc9yycY7XKD8VMlzvVcD9bXz2AlszuLtnPbAB+OIp1LKoupJcAPxn4CNVdWSofc7ndIJ1nTM0+xHgqTZ9H7Cx1bcC2Mh3/4t3SetqtX2AwYeifzLUtpTjNYq9wDXtLp5LgFfbic14xmupPqGexB/w0wyua70OvATc19p/ALh3qN+VwNcYvFN/fKj9/Qz+o5wGfgc4a0x1vRfYDxwEvgCsbO1TwKeG+q1j8O79ruPWvx/4CoPw+i/A902qLuBvt31/uT1ufzuMF/BzwLeAx4b+zl+K8Zrr9cLgctFH2vS72/FPt/F4/9C6H2/rPQ1cMebX+0J1faH9dzA7PnsXek4nVNe/A55o+38A+MDQuv+wjeM0cO0k62rz/wb45HHrLfV4fZbB3WffYpBf24FfAH6hLQ+D/9nUM23/U0PrnvJ4+TMMktSRHi7vSJIaQ1+SOmLoS1JHDH1J6oihL0kdMfQlqSOGviR15P8B5zx2PMItSAcAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.hist(bkgd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "596c0c6a-f8c0-47b1-aac7-83680c1cc324",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
