{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load libraries and Network results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "# -*- coding: utf-8 -*-\n",
    "\"\"\"\n",
    "Created on Fri Sep 25 10:40:42 2020\n",
    "\n",
    "@author: mwall\n",
    "\"\"\"\n",
    "import os\n",
    "import sys\n",
    "import glob\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import miner as miner\n",
    "from scipy.stats import zscore\n",
    "import miner_py3_kk as miner\n",
    "from tqdm.notebook import tqdm, trange\n",
    "import time\n",
    "\n",
    "# change the working directory\n",
    "os.chdir('/Users/serdarturkaslan/Documents/GitHub/GbmMINER/data/MINER_MicroLowessRNATMM.08.24.2020/')\n",
    "\n",
    "# Path to the miner directory\n",
    "input_path = os.getcwd()\n",
    "\n",
    "# Load regulon Modules\n",
    "regulonModules = miner.read_json(os.path.join(input_path,\"regulons.json\"))\n",
    "regulonDfMicroGbmLatest = pd.read_csv(\"regulonDfGbmMicroRNASigCoxAndStatSig.csv\", header = 0)\n",
    "regulonDfMicroGbmLatest = list(regulonDfMicroGbmLatest['Regulon_ID'].drop_duplicates())\n",
    "regulonDfMicroGbmLatest = [str(i) for i in regulonDfMicroGbmLatest]\n",
    "regulonModules = dict((k, regulonModules[k]) for k in regulonDfMicroGbmLatest if k in regulonModules)\n",
    "\n",
    "# create regulon dictionary\n",
    "referenceDictionary = regulonModules"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load data folders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Folders for patient data\n",
    "patientDataFolders = glob.glob('/Volumes/omics4tb2/SYGNAL/XCures/processed_results/*')\n",
    "patientDataFolders = list(filter(lambda x:'TL-' in x, patientDataFolders))\n",
    "print('Total Patients to process: %s' %(len(patientDataFolders)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load Protein coding genes for filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Protein Coding Genes\n",
      "         Gene stable ID Transcript stable ID Gene name       Gene type  \\\n",
      "0       ENSG00000284967      ENST00000647178     FDFT1  protein_coding   \n",
      "1       ENSG00000284967      ENST00000647069     FDFT1  protein_coding   \n",
      "2       ENSG00000079459      ENST00000538689     FDFT1  protein_coding   \n",
      "3       ENSG00000079459      ENST00000615631     FDFT1  protein_coding   \n",
      "4       ENSG00000079459      ENST00000618539     FDFT1  protein_coding   \n",
      "...                 ...                  ...       ...             ...   \n",
      "251334  ENSG00000189171      ENST00000440685   S100A13  protein_coding   \n",
      "251335  ENSG00000189171      ENST00000476133   S100A13  protein_coding   \n",
      "251336  ENSG00000189171      ENST00000339556   S100A13  protein_coding   \n",
      "251337  ENSG00000189171      ENST00000392622   S100A13  protein_coding   \n",
      "251338  ENSG00000189171      ENST00000392623   S100A13  protein_coding   \n",
      "\n",
      "       Gene Synonym  \n",
      "0              DGPT  \n",
      "1              DGPT  \n",
      "2              DGPT  \n",
      "3              DGPT  \n",
      "4              DGPT  \n",
      "...             ...  \n",
      "251334          NaN  \n",
      "251335          NaN  \n",
      "251336          NaN  \n",
      "251337          NaN  \n",
      "251338          NaN  \n",
      "\n",
      "[251339 rows x 5 columns]\n",
      "22492\n"
     ]
    }
   ],
   "source": [
    "# load protein coding genes list from ensembl for filtering\n",
    "ensemblProteinCodingGenes = pd.read_csv(os.path.join(input_path,\"homo_sapiens_protein_coding_export.txt\"), sep=\"\\t\", index_col=None, header = 0)\n",
    "print(\"Protein Coding Genes\")\n",
    "print(ensemblProteinCodingGenes)\n",
    "\n",
    "uniqueProteins = list(ensemblProteinCodingGenes['Gene stable ID'].drop_duplicates())\n",
    "print(len(uniqueProteins))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                target_id  length  eff_length  est_counts  tpm\n",
      "0       ENST00000631435.1      12     6.16667         0.0  0.0\n",
      "1       ENST00000434970.2       9     4.20000         0.0  0.0\n",
      "2       ENST00000448914.1      13     6.28571         0.0  0.0\n",
      "3       ENST00000415118.1       8     3.20000         0.0  0.0\n",
      "4       ENST00000632684.1      12     6.16667         0.0  0.0\n",
      "...                   ...     ...         ...         ...  ...\n",
      "188748  ENST00000639790.1    1370  1135.59000         0.0  0.0\n",
      "188749  ENST00000639660.1     284    90.97150         0.0  0.0\n",
      "188750  ENST00000643577.1     105    18.34970         0.0  0.0\n",
      "188751  ENST00000646356.1     900   665.58700         0.0  0.0\n",
      "188752  ENST00000645792.1     930   695.58700         0.0  0.0\n",
      "\n",
      "[188753 rows x 5 columns]\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Loop through each patient to calculate regulon activity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5d31f2a6604243498236b94b7e02b9e6",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/12 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TL-19-61DB85\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-19-61DB85/RNA/TL-19-61DB85.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-19-87E3E1\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-19-87E3E1/RNA/TL-19-87E3E1.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-20-0B6792\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-20-0B6792/RNA/TL-20-0B6792.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-20-24CC77\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-20-24CC77/RNA/TL-20-24CC77.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-20-495B84\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-20-495B84/RNA/TL-20-495B84.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-20-920D6D\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-20-920D6D/RNA/TL-20-920D6D.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-20-C7EDD0\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-20-C7EDD0/RNA/TL-20-C7EDD0.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-20-E2D448\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-20-E2D448/RNA/TL-20-E2D448.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-20-F66315\n",
      "Data file doesnt exist: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-20-F66315/RNA/TL-20-F66315.genes.results\n",
      "TL-21-3FD45C\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-21-3FD45C/RNA/TL-21-3FD45C.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-19-0B9A1B\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-19-0B9A1B/RNA/TL-19-0B9A1B.genes.results\n",
      "done!\n",
      "done!\n",
      "TL-18-31DCF0\n",
      "Processed: /Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-18-31DCF0/RNA/TL-18-31DCF0.genes.results\n",
      "done!\n",
      "done!\n",
      "              Over  Under  Neutral\n",
      "TL-19-61DB85   257      0      248\n",
      "TL-19-87E3E1   252      0      253\n",
      "TL-20-0B6792   341      0      164\n",
      "TL-20-24CC77   224      0      281\n",
      "TL-20-495B84   268      0      237\n",
      "TL-20-920D6D   202      1      302\n",
      "TL-20-C7EDD0   312      0      193\n",
      "TL-20-E2D448   235      2      268\n",
      "TL-21-3FD45C   233      1      271\n",
      "TL-19-0B9A1B   274      0      231\n",
      "TL-18-31DCF0   330      0      175\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "# Patient analysis loop\n",
    "allSummary = pd.DataFrame()\n",
    "for folder in tqdm(patientDataFolders):\n",
    "    time.sleep(0.01)\n",
    "    patientID = folder.split('/')[-1] #get patientID\n",
    "    print(patientID)\n",
    "    patientDataFile = os.path.join(folder,\"RNA\",patientID + \".genes.results\") # get RNASeq results\n",
    "    \n",
    "    # Check if patient data file exists\n",
    "    if os.path.exists('%s' %(patientDataFile)):\n",
    "        # create a new column with patient name\n",
    "        newColName = patientID + \"_zscore\"\n",
    "        \n",
    "        # Read expression data\n",
    "        rawExpressionData = pd.read_csv(patientDataFile, sep=\"\\t\", index_col=None, header = 0)\n",
    "        \n",
    "        \n",
    "        # seperate ensembl gene ids and symbols\n",
    "        rawExpressionData[['GeneID','gene_symbol']] = pd.DataFrame(rawExpressionData).gene_id.str.split(\"_\",expand=True)\n",
    "        #print(rawExpressionData)\n",
    "        \n",
    "        # filter for protein coding genes\n",
    "        rawExpressionDataFilt = rawExpressionData[rawExpressionData.GeneID.isin(uniqueProteins)].copy()\n",
    "        #rawExpressionDataFilt = rawExpressionData.loc[rawExpressionData.GeneID in uniqueProteins]\n",
    "\n",
    "        #print(\"Filtered raw expression\")\n",
    "        #print(rawExpressionDataFilt)\n",
    "        \n",
    "        # zscore patients expression data\n",
    "        rawExpressionDataFilt[[newColName]] = rawExpressionDataFilt[['TPM']].apply(zscore)\n",
    "   \n",
    "        #expressionData = pd.read_csv(\"abundanceXCuresRNATPMTL1961DB85ZScoredLatest.csv\", sep=\",\", index_col=0, header = 0)\n",
    "        expressionData = rawExpressionDataFilt[[patientID + \"_zscore\"]]\n",
    "        expressionData.index = rawExpressionDataFilt['GeneID']\n",
    "        #print(expressionData)\n",
    "        \n",
    "        print(\"Processed: \" + patientDataFile)\n",
    "        \n",
    "        # create a background matrix used for statistical hypothesis testing\n",
    "        #bkgd = miner.backgroundDf(expressionData)\n",
    "        \n",
    "        # for each cluster, give samples that show high coherent cluster activity\n",
    "        #overExpressedMembers = miner.biclusterMembershipDictionary(referenceDictionary,bkgd,label=2,p=0.05)\n",
    "        \n",
    "        # for each clus|ter, give samples that show low coherent cluster activity\n",
    "        #underExpressedMembers = miner.biclusterMembershipDictionary(referenceDictionary,bkgd,label=0,p=0.05)\n",
    "        \n",
    "        # convert overExpressedMembers dictionary to binary matrix\n",
    "        #overExpressedMembersMatrix = miner.membershipToIncidence(overExpressedMembers,expressionData)\n",
    "        #overExpressedMembersMatrix.index = np.array(overExpressedMembersMatrix.index).astype(str)\n",
    "        \n",
    "        # convert underExpressedMembers dictionary to binary matrix\n",
    "        #underExpressedMembersMatrix = miner.membershipToIncidence(underExpressedMembers,expressionData)\n",
    "        #underExpressedMembersMatrix.index = np.array(underExpressedMembersMatrix.index).astype(str)\n",
    "\n",
    "        # Create program matrix with values of {-1,0,1}\n",
    "        #dfr_regulons = overExpressedMembersMatrix-underExpressedMembersMatrix\n",
    "        #print(dfr_regulons)\n",
    "        # \n",
    "        \n",
    "        # calculate regulon activity\n",
    "        rr = miner.generateRegulonActivity(referenceDictionary,expressionData, p=0.05)\n",
    "        #print(rr)\n",
    "        \n",
    "        # write regulon activity results to file\n",
    "        rr.to_csv(\"Regulon_Activity_\" + patientID + \".csv\")\n",
    "        # calculate regulon activity stats\n",
    "        overActiveRegulonCount = len(rr[rr[newColName] == 1])\n",
    "        underActiveRegulonCount = len(rr[rr[newColName] == -1])\n",
    "        neutralRegulonCount = len(rr[rr[newColName] == 0])\n",
    "        \n",
    "        resSummary = {\"Over\" : [overActiveRegulonCount],\n",
    "                              \"Under\" : [underActiveRegulonCount],\n",
    "                              \"Neutral\" : [neutralRegulonCount]}\n",
    "        resSummary = pd.DataFrame(resSummary)\n",
    "        resSummary = resSummary.rename(index={0: patientID})\n",
    "        \n",
    "        \n",
    "        allSummary = allSummary.append(resSummary)\n",
    "        \n",
    "        #print(resSummary)\n",
    "        \n",
    "        #dfr_regulons = miner.generateRegulonActivity(referenceDictionary,expressionData, p=0.05, returnBkgd=\"no\")\n",
    "        \n",
    "#         overActiveRegulonCount = len(dfr_regulons[dfr_regulons[patientID+\"_zscore\"] == 1])\n",
    "#         underActiveRegulonCount = len(dfr_regulons[dfr_regulons[patientID+\"_zscore\"] == -1])\n",
    "#         neutralRegulonCount = len(dfr_regulons[dfr_regulons[patientID+\"_zscore\"] == 0])\n",
    "        \n",
    "#         resSummary = {\"Over\" : [overActiveRegulonCount],\n",
    "#                       \"Under\" : [underActiveRegulonCount],\n",
    "#                       \"Neutral\" : [neutralRegulonCount]}\n",
    "#         resSummary = pd.DataFrame(resSummary)\n",
    "#         resSummary = resSummary.rename(index={0: str(patientID)})\n",
    "#         allSummary = allSummary.append(resSummary)\n",
    "        #print(resSummary)\n",
    "\n",
    "        #print(\"ActiveactiveRegulonCount:\", activeRegulonCount)\n",
    "        #sys.exit()\n",
    "\n",
    "        #dfr_regulons.to_csv(\"Regulon_Activity_\" + patientID + \".csv\")\n",
    "        \n",
    "    else:\n",
    "        print(\"Data file doesnt exist: \" + patientDataFile) \n",
    "print(allSummary)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "target_id\n",
      "ENST00000631435.1    0.0\n",
      "ENST00000434970.2    0.0\n",
      "ENST00000448914.1    0.0\n",
      "ENST00000415118.1    0.0\n",
      "ENST00000632684.1    0.0\n",
      "                    ... \n",
      "ENST00000639790.1    0.0\n",
      "ENST00000639660.1    0.0\n",
      "ENST00000643577.1    0.0\n",
      "ENST00000646356.1    0.0\n",
      "ENST00000645792.1    0.0\n",
      "Name: tpm, Length: 188753, dtype: float64\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "object of type 'numpy.float64' has no len()",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/dy/kk9lybnx3m56nwpxv_wjhjrw0000gq/T/ipykernel_44119/1499675574.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexpressionData\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;31m#miner.generateRegulonActivity()\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m \u001b[0mrr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mminer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgenerateRegulonActivity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreferenceDictionary\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mexpressionData\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0.025\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/GitHub/GbmMINER/src/miner_py3_kk.py\u001b[0m in \u001b[0;36mgenerateRegulonActivity\u001b[0;34m(regulonModules, expressionData, p, returnBkgd)\u001b[0m\n\u001b[1;32m   5749\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5750\u001b[0m     \u001b[0;31m# create a background matrix used for statistical hypothesis testing\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 5751\u001b[0;31m     \u001b[0mbkgd\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbackgroundDf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexpressionData\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5752\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5753\u001b[0m     \u001b[0;31m# for each cluster, give samples that show high coherent cluster activity\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/GitHub/GbmMINER/src/miner_py3_kk.py\u001b[0m in \u001b[0;36mbackgroundDf\u001b[0;34m(expressionData)\u001b[0m\n\u001b[1;32m    967\u001b[0m     \u001b[0mlow\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpercentile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexpressionData\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m100.\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    968\u001b[0m     \u001b[0mhigh\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpercentile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexpressionData\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m200.\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 969\u001b[0;31m     \u001b[0mevenCuts\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mzipper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mlow\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mhigh\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    970\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    971\u001b[0m     \u001b[0mbkgd\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mexpressionData\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/GitHub/GbmMINER/src/miner_py3_kk.py\u001b[0m in \u001b[0;36mzipper\u001b[0;34m(ls)\u001b[0m\n\u001b[1;32m   1288\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mzipper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mls\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1289\u001b[0m     \u001b[0mzipped\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1290\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mls\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1291\u001b[0m         \u001b[0mvals\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1292\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mj\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mls\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: object of type 'numpy.float64' has no len()"
     ]
    }
   ],
   "source": [
    "import miner_py3_kk as miner\n",
    "\n",
    "#expressionData = pd.read_csv(\"abundanceXCuresRNATPMTL1961DB85ZScoredLatest.csv\", sep=\",\", index_col=0, header = 0)\n",
    "expressionData = pd.read_csv(\"/Volumes/omics4tb2/SYGNAL/XCures/TL-20-0B6792/RNA/results_kallisto/abundance.tsv\", sep=\"\\t\", index_col=None, header = 0)\n",
    "\n",
    "print(expressionData)\n",
    "#miner.generateRegulonActivity()\n",
    "rr = miner.generateRegulonActivity(referenceDictionary,expressionData, p=0.025)\n",
    "print(rr)\n",
    "\n",
    "overActiveRegulonCount = len(rr[rr['TL1961DB85'] == 1])\n",
    "underActiveRegulonCount = len(rr[rr['TL1961DB85'] == -1])\n",
    "neutralRegulonCount = len(rr[rr['TL1961DB85'] == 0])\n",
    "resSummary = {\"Over\" : [overActiveRegulonCount],\n",
    "                      \"Under\" : [underActiveRegulonCount],\n",
    "                      \"Neutral\" : [neutralRegulonCount]}\n",
    "resSummary = pd.DataFrame(resSummary)\n",
    "resSummary = resSummary.rename(index={0: 'TL1961DB85'})\n",
    "print(resSummary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                          gene_id  \\\n",
      "0          ENSG00000000003_TSPAN6   \n",
      "1            ENSG00000000005_TNMD   \n",
      "2            ENSG00000000419_DPM1   \n",
      "3           ENSG00000000457_SCYL3   \n",
      "4        ENSG00000000460_C1orf112   \n",
      "...                           ...   \n",
      "60671    ENSG00000288584_Z97205.3   \n",
      "60672  ENSG00000288585_AC010184.1   \n",
      "60673  ENSG00000288586_AL357874.3   \n",
      "60674  ENSG00000288587_AL645933.5   \n",
      "60675  ENSG00000288588_AC116317.2   \n",
      "\n",
      "                                        transcript_id(s)   length  \\\n",
      "0      ENST00000373020_TSPAN6-201,ENST00000494424_TSP...  2547.10   \n",
      "1      ENST00000373031_TNMD-201,ENST00000485971_TNMD-202  1205.00   \n",
      "2      ENST00000371582_DPM1-201,ENST00000371584_DPM1-...  1075.97   \n",
      "3      ENST00000367770_SCYL3-201,ENST00000367771_SCYL...  4091.68   \n",
      "4      ENST00000286031_C1orf112-201,ENST00000359326_C...  2103.71   \n",
      "...                                                  ...      ...   \n",
      "60671                       ENST00000655888_Z97205.3-201  2595.00   \n",
      "60672                     ENST00000673707_AC010184.1-201  5800.00   \n",
      "60673                     ENST00000673624_AL357874.3-201  1703.00   \n",
      "60674                     ENST00000673857_AL645933.5-201   554.00   \n",
      "60675                     ENST00000673884_AC116317.2-201  1086.00   \n",
      "\n",
      "       effective_length  expected_count    TPM   FPKM  \n",
      "0               2302.51         1062.00  21.84  15.51  \n",
      "1                960.41            4.00   0.20   0.14  \n",
      "2                831.39          820.00  46.71  33.16  \n",
      "3               3847.10          729.71   8.98   6.38  \n",
      "4               1859.15          861.29  21.94  15.58  \n",
      "...                 ...             ...    ...    ...  \n",
      "60671           2350.41            0.00   0.00   0.00  \n",
      "60672           5555.41            0.00   0.00   0.00  \n",
      "60673           1458.41            0.00   0.00   0.00  \n",
      "60674            310.03            0.00   0.00   0.00  \n",
      "60675            841.41            0.00   0.00   0.00  \n",
      "\n",
      "[60676 rows x 7 columns]\n",
      "60676\n",
      "done!\n",
      "done!\n",
      "      zscore\n",
      "0        1.0\n",
      "4        1.0\n",
      "6        1.0\n",
      "14       1.0\n",
      "23       1.0\n",
      "...      ...\n",
      "3675     1.0\n",
      "3703     1.0\n",
      "3707     1.0\n",
      "3727     1.0\n",
      "3732     1.0\n",
      "\n",
      "[505 rows x 1 columns]\n",
      "        Over  Under  Neutral\n",
      "zscore   505      0        0\n"
     ]
    }
   ],
   "source": [
    "import miner_py3_kk as miner\n",
    "from scipy.stats import zscore\n",
    "\n",
    "\n",
    "rawExpressionData = pd.read_csv(\"/Volumes/omics4tb2/SYGNAL/XCures/processed_results/TL-19-87E3E1/RNA/TL-19-87E3E1.genes.results\", sep=\"\\t\", index_col=None, header = 0)\n",
    "print(rawExpressionData)\n",
    "# zscore patients expression data\n",
    "rawExpressionData[[\"zscore\"]] = rawExpressionData[['TPM']].apply(zscore)\n",
    "\n",
    "# seperate ensembl gene ids and symbols\n",
    "rawExpressionData[['GeneID','gene_symbol']] = pd.DataFrame(rawExpressionData).gene_id.str.split(\"_\",expand=True)\n",
    "print(len(set(rawExpressionData['GeneID'])))\n",
    "\n",
    "#expressionData = pd.read_csv(\"abundanceXCuresRNATPMTL1961DB85ZScoredLatest.csv\", sep=\",\", index_col=0, header = 0)\n",
    "expressionData = rawExpressionData[[\"zscore\"]]\n",
    "expressionData.index = rawExpressionData['GeneID']\n",
    "\n",
    "\n",
    "#miner.generateRegulonActivity()\n",
    "rr = miner.generateRegulonActivity(referenceDictionary,expressionData, p=0.025)\n",
    "print(rr)\n",
    "\n",
    "overActiveRegulonCount = len(rr[rr['zscore'] == 1])\n",
    "underActiveRegulonCount = len(rr[rr['zscore'] == -1])\n",
    "neutralRegulonCount = len(rr[rr['zscore'] == 0])\n",
    "resSummary = {\"Over\" : [overActiveRegulonCount],\n",
    "                      \"Under\" : [underActiveRegulonCount],\n",
    "                      \"Neutral\" : [neutralRegulonCount]}\n",
    "resSummary = pd.DataFrame(resSummary)\n",
    "resSummary = resSummary.rename(index={0: 'zscore'})\n",
    "print(resSummary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done!\n",
      "done!\n",
      "      TL1961DB85\n",
      "0            0.0\n",
      "4            0.0\n",
      "6            1.0\n",
      "14           0.0\n",
      "23           0.0\n",
      "...          ...\n",
      "3675         0.0\n",
      "3703         0.0\n",
      "3707         0.0\n",
      "3727         1.0\n",
      "3732         0.0\n",
      "\n",
      "[505 rows x 1 columns]\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# create a background matrix used for statistical hypothesis testing\n",
    "bkgd = miner.backgroundDf(expressionData)\n",
    "# for each cluster, give samples that show high coherent cluster activity\n",
    "overExpressedMembers = miner.biclusterMembershipDictionary(referenceDictionary,bkgd,label=2,p=0.05)\n",
    "# for each clus|ter, give samples that show low coherent cluster activity\n",
    "underExpressedMembers = miner.biclusterMembershipDictionary(referenceDictionary,bkgd,label=0,p=0.05)\n",
    "# convert overExpressedMembers dictionary to binary matrix\n",
    "overExpressedMembersMatrix = miner.membershipToIncidence(overExpressedMembers,expressionData)\n",
    "overExpressedMembersMatrix.index = np.array(overExpressedMembersMatrix.index).astype(str)\n",
    "# convert underExpressedMembers dictionary to binary matrix\n",
    "underExpressedMembersMatrix = miner.membershipToIncidence(underExpressedMembers,expressionData)\n",
    "underExpressedMembersMatrix.index = np.array(underExpressedMembersMatrix.index).astype(str)\n",
    "\n",
    "# Create program matrix with values of {-1,0,1}\n",
    "dfr_regulons = overExpressedMembersMatrix-underExpressedMembersMatrix\n",
    "print(dfr_regulons)\n",
    "\n",
    "dfr_regulons.to_csv(\"SigRegulonsActivityabundanceXCuresRNATPMTL1961DB85ZScoredLatest.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
